{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "id": "TLQwwG5VDPFA"
   },
   "outputs": [],
   "source": [
    "import pandas\n",
    "import numpy\n",
    "import scipy\n",
    "import math\n",
    "import scipy.stats\n",
    "import seaborn"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "CMOWiLBfDPFA"
   },
   "outputs": [],
   "source": [
    "# module for figures\n",
    "import matplotlib, matplotlib.pyplot as plt\n",
    "\n",
    "# this is a trick to make figures look nicer\n",
    "matplotlib.rcParams.update({'font.size':20, 'font.family':'FreeSans', 'xtick.labelsize':20, 'ytick.labelsize':10, 'figure.figsize':(12, 8)})"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "aHG0vowTDPFC"
   },
   "source": [
    "# 0. user-defined variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "input_file_directory = '/home/adrian/projects/ATG7/data/'\n",
    "\n",
    "normal_tissue_expression_file = '/home/adrian/projects/ATG7/results/normal_tissue_expression.tsv'\n",
    "primary_tumor_expression_file = '/home/adrian/projects/ATG7/results/primary_tumors_expression.tsv'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "wYGSc-xPDaFV"
   },
   "source": [
    "## functions & options\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "id": "jmnIMX-FfoyD"
   },
   "outputs": [],
   "source": [
    "pandas.set_option(\"max_rows\", None)\n",
    "# pandas.reset_option(\"max_rows\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "id": "K8VbSwcZyDon"
   },
   "outputs": [],
   "source": [
    "#define a function to split after the \".\". The 1 is for how many word to have. if we have several dot. The [1] is what is after the dot, and [0] what is before\n",
    "def split(L):\n",
    "  return L.split(\".\",1)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "id": "ccNktmiu2K2n"
   },
   "outputs": [],
   "source": [
    "def binarise_mutation(DF):\n",
    "  old = ['no variant', \"3'UTR\", \"5'UTR\", 'Intron', \"5'Flank\", 'Silent', \n",
    "         'large deletion', 'Nonstop_Mutation','Missense_Mutation', 'Nonsense_Mutation', 'Frame_Shift_Del', 'Frame_Shift_Ins', 'Splice_Site', 'In_Frame_Del', 'In_Frame_Ins', 'Translation_Start_Site']\n",
    "  new = [0,0,0,0,0,0,\n",
    "         1,1,1,1,1,1,1,1,1,1]\n",
    "\n",
    "  for i in range(len(old)):\n",
    "    DF['effect'] = DF['effect'].replace(old[i], new[i])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "IpmCT5mEDPFD"
   },
   "source": [
    "# II] Dataframe settings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "scwtwuDCqUKz"
   },
   "source": [
    "## A) Isoform data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "id": "PHqx_s0JozXU"
   },
   "outputs": [],
   "source": [
    "# Original data isoforms\n",
    "directory = input_file_directory + \"xena_surv_ATG7.tsv\"\n",
    "df_ori= pandas.read_csv(directory, sep = \"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "oC8hQXN9WRsH",
    "outputId": "17c60649-f5dd-4b4d-bb08-d6c2e32a1332",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  # This is added back by InteractiveShellApp.init_path()\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:5047: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:30: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:31: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:34: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:35: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:40: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ATG7_3</th>\n",
       "      <th>ATG7(total)</th>\n",
       "      <th>ATG7(1)</th>\n",
       "      <th>ATG7(2)</th>\n",
       "      <th>ATG7(3)</th>\n",
       "      <th>log2_+1_ATG7(total)</th>\n",
       "      <th>log2_+1_ATG7(1)</th>\n",
       "      <th>log2_+1_ATG7(2)</th>\n",
       "      <th>ATG7_prot_tot</th>\n",
       "      <th>ATG7_1%</th>\n",
       "      <th>ATG7_2%</th>\n",
       "      <th>ATG7_3%</th>\n",
       "      <th>ATG7(1)norm</th>\n",
       "      <th>ATG7(2)norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TARGET-20-PARUBT-40</td>\n",
       "      <td>Recurrent Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>4.401</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>1.522</td>\n",
       "      <td>-1.7320</td>\n",
       "      <td>21.125765</td>\n",
       "      <td>2.619787</td>\n",
       "      <td>2.870889</td>\n",
       "      <td>3.000343e-01</td>\n",
       "      <td>4.467655</td>\n",
       "      <td>1.855905</td>\n",
       "      <td>1.952665</td>\n",
       "      <td>5.790710</td>\n",
       "      <td>45.241201</td>\n",
       "      <td>49.577495</td>\n",
       "      <td>5.181305</td>\n",
       "      <td>0.124009</td>\n",
       "      <td>0.135895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TARGET-20-PATJHJ-40</td>\n",
       "      <td>Recurrent Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>2.820</td>\n",
       "      <td>1.1180</td>\n",
       "      <td>-1.086</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>7.060624</td>\n",
       "      <td>2.169459</td>\n",
       "      <td>0.470066</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>3.010892</td>\n",
       "      <td>1.664236</td>\n",
       "      <td>0.555881</td>\n",
       "      <td>2.639524</td>\n",
       "      <td>82.191280</td>\n",
       "      <td>17.808726</td>\n",
       "      <td>-0.000006</td>\n",
       "      <td>0.307262</td>\n",
       "      <td>0.066576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TARGET-10-PASLZM-40</td>\n",
       "      <td>Recurrent Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>1.158</td>\n",
       "      <td>0.0158</td>\n",
       "      <td>-3.171</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>2.230479</td>\n",
       "      <td>1.010012</td>\n",
       "      <td>0.110028</td>\n",
       "      <td>7.000183e-02</td>\n",
       "      <td>1.691748</td>\n",
       "      <td>1.007204</td>\n",
       "      <td>0.150597</td>\n",
       "      <td>1.190042</td>\n",
       "      <td>84.871949</td>\n",
       "      <td>9.245753</td>\n",
       "      <td>5.882299</td>\n",
       "      <td>0.452823</td>\n",
       "      <td>0.049329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TARGET-21-PATAIJ-42</td>\n",
       "      <td>Post treatment Blood Cancer - Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>4.332</td>\n",
       "      <td>1.3450</td>\n",
       "      <td>1.975</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>20.139115</td>\n",
       "      <td>2.539302</td>\n",
       "      <td>3.930282</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>4.401843</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>2.301670</td>\n",
       "      <td>6.469584</td>\n",
       "      <td>39.249848</td>\n",
       "      <td>60.750154</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>0.126088</td>\n",
       "      <td>0.195157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TARGET-21-PASVJS-41</td>\n",
       "      <td>Post treatment Blood Cancer - Bone Marrow</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>5.156</td>\n",
       "      <td>1.8040</td>\n",
       "      <td>3.769</td>\n",
       "      <td>0.5069</td>\n",
       "      <td>35.653197</td>\n",
       "      <td>3.490870</td>\n",
       "      <td>13.631706</td>\n",
       "      <td>1.419994e+00</td>\n",
       "      <td>5.195867</td>\n",
       "      <td>2.166995</td>\n",
       "      <td>3.871026</td>\n",
       "      <td>18.542569</td>\n",
       "      <td>18.826249</td>\n",
       "      <td>73.515731</td>\n",
       "      <td>7.658019</td>\n",
       "      <td>0.097912</td>\n",
       "      <td>0.382342</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample                                        Sample_Type  \\\n",
       "0  TARGET-20-PARUBT-40  Recurrent Blood Derived Cancer - Peripheral Blood   \n",
       "1  TARGET-20-PATJHJ-40  Recurrent Blood Derived Cancer - Peripheral Blood   \n",
       "2  TARGET-10-PASLZM-40  Recurrent Blood Derived Cancer - Peripheral Blood   \n",
       "3  TARGET-21-PATAIJ-42                Post treatment Blood Cancer - Blood   \n",
       "4  TARGET-21-PASVJS-41          Post treatment Blood Cancer - Bone Marrow   \n",
       "\n",
       "       Primary_Site   ATG7  ATG7_1  ATG7_2  ATG7_3  ATG7(total)   ATG7(1)  \\\n",
       "0  White Blood Cell  4.401  1.3900   1.522 -1.7320    21.125765  2.619787   \n",
       "1  White Blood Cell  2.820  1.1180  -1.086 -9.9660     7.060624  2.169459   \n",
       "2  White Blood Cell  1.158  0.0158  -3.171 -3.8160     2.230479  1.010012   \n",
       "3  White Blood Cell  4.332  1.3450   1.975 -9.9660    20.139115  2.539302   \n",
       "4  White Blood Cell  5.156  1.8040   3.769  0.5069    35.653197  3.490870   \n",
       "\n",
       "     ATG7(2)       ATG7(3)  log2_+1_ATG7(total)  log2_+1_ATG7(1)  \\\n",
       "0   2.870889  3.000343e-01             4.467655         1.855905   \n",
       "1   0.470066 -1.495113e-07             3.010892         1.664236   \n",
       "2   0.110028  7.000183e-02             1.691748         1.007204   \n",
       "3   3.930282 -1.495113e-07             4.401843         1.823465   \n",
       "4  13.631706  1.419994e+00             5.195867         2.166995   \n",
       "\n",
       "   log2_+1_ATG7(2)  ATG7_prot_tot    ATG7_1%    ATG7_2%   ATG7_3%  \\\n",
       "0         1.952665       5.790710  45.241201  49.577495  5.181305   \n",
       "1         0.555881       2.639524  82.191280  17.808726 -0.000006   \n",
       "2         0.150597       1.190042  84.871949   9.245753  5.882299   \n",
       "3         2.301670       6.469584  39.249848  60.750154 -0.000002   \n",
       "4         3.871026      18.542569  18.826249  73.515731  7.658019   \n",
       "\n",
       "   ATG7(1)norm  ATG7(2)norm  \n",
       "0     0.124009     0.135895  \n",
       "1     0.307262     0.066576  \n",
       "2     0.452823     0.049329  \n",
       "3     0.126088     0.195157  \n",
       "4     0.097912     0.382342  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Column selection\n",
    "df= df_ori[['sample',\n",
    "            '_sample_type',\n",
    "            '_primary_site',\n",
    "            'ENSG00000197548.12',\n",
    "            'ENST00000354449.7', \n",
    "            'ENST00000354956.9',\n",
    "            'ENST00000446450.6']]\n",
    "\n",
    "# We have Adrenal Gland and Adrenal gland. I need to put \"G\".\n",
    "df['_primary_site'] = df['_primary_site'].str.title()\n",
    "\n",
    "# Column rename\n",
    "df.rename(columns = {'sample' : 'sample',\n",
    "    '_sample_type' : 'Sample_Type', \n",
    "                            '_primary_site' : 'Primary_Site', \n",
    "                            'ENST00000354449.7': 'ATG7_1',\n",
    "                            'ENST00000354956.9': 'ATG7_2',\n",
    "                            'ENST00000446450.6': 'ATG7_3',\n",
    "                            'ENSG00000197548.12': 'ATG7'}, inplace = True)\n",
    "\n",
    "# Calcul to have value from log2;  2**(ATG7(1))-0.001\n",
    "df[\"ATG7(total)\"] = pow(2,df['ATG7'])-0.001\n",
    "df[\"ATG7(1)\"] = pow(2,df['ATG7_1'])-0.001\n",
    "df[\"ATG7(2)\"] = pow(2,df['ATG7_2'])-0.001\n",
    "df[\"ATG7(3)\"] = pow(2,df['ATG7_3'])-0.001\n",
    "\n",
    "# Calcul to have log2+1 from value; \n",
    "df[\"log2_+1_ATG7(total)\"] = numpy.log2(df['ATG7(total)']+1)\n",
    "df[\"log2_+1_ATG7(1)\"] = numpy.log2(df['ATG7(1)']+1)\n",
    "df[\"log2_+1_ATG7(2)\"] = numpy.log2(df['ATG7(2)']+1)\n",
    "\n",
    "# Calcul to see percentage of expression of the 3 isoforms of ATG7\n",
    "df['ATG7_prot_tot'] = (df['ATG7(1)'] + df['ATG7(2)'] + df['ATG7(3)'])\n",
    "df['ATG7_1%'] = (df['ATG7(1)'] *100)/ df['ATG7_prot_tot']\n",
    "df['ATG7_2%'] = (df['ATG7(2)'] *100)/ df['ATG7_prot_tot']\n",
    "df['ATG7_3%'] = (df['ATG7(3)'] *100)/ df['ATG7_prot_tot']\n",
    "\n",
    "#Add columns, normalized on total\n",
    "df[\"ATG7(1)norm\"] = df['ATG7(1)']/df[\"ATG7(total)\"]\n",
    "df[\"ATG7(2)norm\"] = df['ATG7(2)']/df[\"ATG7(total)\"]\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "l2tWUsTWX8ES",
    "outputId": "1ed530fb-9179-4afd-a95f-2d36353eef32"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Primary Tumor                                        9185\n",
       "Normal Tissue                                        7429\n",
       "Solid Tissue Normal                                   738\n",
       "Cell Line                                             433\n",
       "Metastatic                                            393\n",
       "Primary Solid Tumor                                   286\n",
       "Primary Blood Derived Cancer - Peripheral Blood       239\n",
       "Primary Blood Derived Cancer - Bone Marrow            237\n",
       "Recurrent Blood Derived Cancer - Bone Marrow          104\n",
       "Recurrent Tumor                                        45\n",
       "Recurrent Solid Tumor                                  13\n",
       "Post treatment Blood Cancer - Bone Marrow              12\n",
       "Additional - New Primary                               11\n",
       "Recurrent Blood Derived Cancer - Peripheral Blood       3\n",
       "Post treatment Blood Cancer - Blood                     1\n",
       "Control Analyte                                         1\n",
       "Additional Metastatic                                   1\n",
       "Name: Sample_Type, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Sample_Type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "kteWDMcBRLe1",
    "outputId": "609ba62e-868c-4942-dd33-728803f4ca7c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "normal_tissue 7429\n",
      "primary_tumors 9185\n"
     ]
    }
   ],
   "source": [
    "#Creation of two tables, Normal tissue and Primary tumor.\n",
    "df_ATG7_Normal = df[df['Sample_Type'].str.match ('Normal Tissue')]\n",
    "df_ATG7_Primary = df[df['Sample_Type'].str.match ('Primary Tumor')]\n",
    "\n",
    "print('normal_tissue' , len(df_ATG7_Normal))\n",
    "print('primary_tumors' , len(df_ATG7_Primary))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 366
    },
    "id": "90ZnSHOcSa--",
    "outputId": "a04c738c-d866-42e0-ff86-5ae0fce970ac"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ATG7_3</th>\n",
       "      <th>ATG7(total)</th>\n",
       "      <th>ATG7(1)</th>\n",
       "      <th>ATG7(2)</th>\n",
       "      <th>ATG7(3)</th>\n",
       "      <th>log2_+1_ATG7(total)</th>\n",
       "      <th>log2_+1_ATG7(1)</th>\n",
       "      <th>log2_+1_ATG7(2)</th>\n",
       "      <th>ATG7_prot_tot</th>\n",
       "      <th>ATG7_1%</th>\n",
       "      <th>ATG7_2%</th>\n",
       "      <th>ATG7_3%</th>\n",
       "      <th>ATG7(1)norm</th>\n",
       "      <th>ATG7(2)norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GTEX-ZTTD-0326-SM-57WFW</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>5.042</td>\n",
       "      <td>3.860</td>\n",
       "      <td>2.390</td>\n",
       "      <td>0.2642</td>\n",
       "      <td>32.944283</td>\n",
       "      <td>14.519306</td>\n",
       "      <td>5.240574</td>\n",
       "      <td>1.199970e+00</td>\n",
       "      <td>5.085097</td>\n",
       "      <td>3.955992</td>\n",
       "      <td>2.641679</td>\n",
       "      <td>20.959850</td>\n",
       "      <td>69.271996</td>\n",
       "      <td>25.002916</td>\n",
       "      <td>5.725088</td>\n",
       "      <td>0.440723</td>\n",
       "      <td>0.159074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GTEX-PX3G-1626-SM-2S1PT</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>4.252</td>\n",
       "      <td>2.322</td>\n",
       "      <td>1.956</td>\n",
       "      <td>1.1320</td>\n",
       "      <td>19.052710</td>\n",
       "      <td>4.999249</td>\n",
       "      <td>3.878848</td>\n",
       "      <td>2.190624e+00</td>\n",
       "      <td>4.325725</td>\n",
       "      <td>2.584782</td>\n",
       "      <td>2.286540</td>\n",
       "      <td>11.068720</td>\n",
       "      <td>45.165557</td>\n",
       "      <td>35.043325</td>\n",
       "      <td>19.791118</td>\n",
       "      <td>0.262390</td>\n",
       "      <td>0.203585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GTEX-OXRO-1726-SM-3LK6C</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.909</td>\n",
       "      <td>2.602</td>\n",
       "      <td>1.345</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>15.020948</td>\n",
       "      <td>6.070277</td>\n",
       "      <td>2.539302</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>4.001888</td>\n",
       "      <td>2.821767</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>8.609579</td>\n",
       "      <td>70.506086</td>\n",
       "      <td>29.493916</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>0.404121</td>\n",
       "      <td>0.169051</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample    Sample_Type Primary_Site   ATG7  ATG7_1  ATG7_2  \\\n",
       "0  GTEX-ZTTD-0326-SM-57WFW  Normal Tissue       Muscle  5.042   3.860   2.390   \n",
       "1  GTEX-PX3G-1626-SM-2S1PT  Normal Tissue       Muscle  4.252   2.322   1.956   \n",
       "2  GTEX-OXRO-1726-SM-3LK6C  Normal Tissue       Muscle  3.909   2.602   1.345   \n",
       "\n",
       "   ATG7_3  ATG7(total)    ATG7(1)   ATG7(2)       ATG7(3)  \\\n",
       "0  0.2642    32.944283  14.519306  5.240574  1.199970e+00   \n",
       "1  1.1320    19.052710   4.999249  3.878848  2.190624e+00   \n",
       "2 -9.9660    15.020948   6.070277  2.539302 -1.495113e-07   \n",
       "\n",
       "   log2_+1_ATG7(total)  log2_+1_ATG7(1)  log2_+1_ATG7(2)  ATG7_prot_tot  \\\n",
       "0             5.085097         3.955992         2.641679      20.959850   \n",
       "1             4.325725         2.584782         2.286540      11.068720   \n",
       "2             4.001888         2.821767         1.823465       8.609579   \n",
       "\n",
       "     ATG7_1%    ATG7_2%    ATG7_3%  ATG7(1)norm  ATG7(2)norm  \n",
       "0  69.271996  25.002916   5.725088     0.440723     0.159074  \n",
       "1  45.165557  35.043325  19.791118     0.262390     0.203585  \n",
       "2  70.506086  29.493916  -0.000002     0.404121     0.169051  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df with only Normal and Primary \n",
    "df_Norm_Prim = pandas.concat([df_ATG7_Normal, df_ATG7_Primary], axis=0)\n",
    "df_Norm_Prim.reset_index(drop=True, inplace=True)\n",
    "df_Norm_Prim.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "CGVz9_mqWgpw"
   },
   "source": [
    "### High/Low expression per Median for Isoform expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "id": "Ypc3K2bKWzpI"
   },
   "outputs": [],
   "source": [
    "# What is the median for iso 1 and iso 2?\n",
    "#High >50% and low <50%"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "id": "oA7fE5E3pxi_"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  # This is added back by InteractiveShellApp.init_path()\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "Data = [[df_ATG7_Normal, 'Normal'], [df_ATG7_Primary, 'Primary']]\n",
    "isoforms = [[\"ATG7_1\", \"iso1\"], [\"ATG7_2\", \"iso2\"]]\n",
    "\n",
    "df_LoHi = [] \n",
    "\n",
    "for df in Data:\n",
    "  for isoform in isoforms:\n",
    "    xMed = df[0][isoform[0]].median()\n",
    "\n",
    "    low = df[0].loc[df[0][isoform[0]] < xMed]\n",
    "    low['L/H'] = \"Low_\"+isoform[1]\n",
    "\n",
    "    high = df[0].loc[df[0][isoform[0]] > xMed]\n",
    "    high['L/H'] = \"High_\"+isoform[1]\n",
    "\n",
    "  #merge low and high\n",
    "  concat = pandas.concat([low, high], axis=0)\n",
    "  col = concat.pop('L/H')\n",
    "  concat.insert(3, 'L/H', col)\n",
    "  df_LoHi.append(concat)\n",
    "\n",
    "normal_LoHi = df_LoHi[0]\n",
    "primary_LoHi = df_LoHi[1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "fbjL51C8RltI"
   },
   "source": [
    "## B) Whole gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "id": "dpYmq2ernzq8"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22796\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>entrezgene_id</th>\n",
       "      <th>transcript_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>4535.0</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>4536.0</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ensembl_gene_id hgnc_symbol  entrezgene_id transcript_biotype\n",
       "1  ENSG00000198888      MT-ND1         4535.0     protein_coding\n",
       "2  ENSG00000198763      MT-ND2         4536.0     protein_coding"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Import list of protein coding transcripts\n",
    "protein_coding_directory = input_file_directory + \"protein_coding_list.txt\"\n",
    "protein_coding_list= pandas.read_csv(protein_coding_directory, sep = \"\\t\")\n",
    "print(len(protein_coding_list))\n",
    "protein_coding_list.head(2)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "UcSgLCHbGUmc"
   },
   "source": []
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "id": "cF-WsBgTxw_W"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 2min 27s, sys: 3.4 s, total: 2min 31s\n",
      "Wall time: 2min 31s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "df_gexpr_normal= pandas.read_csv(normal_tissue_expression_file, sep = \"\\t\")\n",
    "df_gexpr_primary= pandas.read_csv(primary_tumor_expression_file, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-1117F-0426-SM-5EGHI</th>\n",
       "      <th>GTEX-1117F-0526-SM-5EGHJ</th>\n",
       "      <th>GTEX-1117F-0626-SM-5N9CS</th>\n",
       "      <th>GTEX-1117F-0726-SM-5GIEN</th>\n",
       "      <th>GTEX-1117F-1326-SM-5EGHH</th>\n",
       "      <th>GTEX-1117F-2226-SM-5N9CH</th>\n",
       "      <th>GTEX-1117F-2426-SM-5EGGH</th>\n",
       "      <th>GTEX-1117F-2826-SM-5GZXL</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZZPU-0726-SM-5N9C8</th>\n",
       "      <th>GTEX-ZZPU-0926-SM-5GZYT</th>\n",
       "      <th>GTEX-ZZPU-1226-SM-5N9CK</th>\n",
       "      <th>GTEX-ZZPU-1326-SM-5GZWS</th>\n",
       "      <th>GTEX-ZZPU-1426-SM-5GZZ6</th>\n",
       "      <th>GTEX-ZZPU-1826-SM-5E43L</th>\n",
       "      <th>GTEX-ZZPU-2126-SM-5EGIU</th>\n",
       "      <th>GTEX-ZZPU-2226-SM-5EGIV</th>\n",
       "      <th>GTEX-ZZPU-2626-SM-5E45Y</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000242268.2</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-1.2481</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>-1.7809</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.8084</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000259041.1</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000270112.3</td>\n",
       "      <td>-4.2934</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-5.5735</td>\n",
       "      <td>0.3573</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-6.5064</td>\n",
       "      <td>-5.0116</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-1.8314</td>\n",
       "      <td>-1.7809</td>\n",
       "      <td>-2.0529</td>\n",
       "      <td>-6.5064</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-2.8262</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000167578.16</td>\n",
       "      <td>5.1190</td>\n",
       "      <td>4.1277</td>\n",
       "      <td>4.4067</td>\n",
       "      <td>5.6860</td>\n",
       "      <td>4.0357</td>\n",
       "      <td>4.6849</td>\n",
       "      <td>4.5009</td>\n",
       "      <td>5.3954</td>\n",
       "      <td>4.9402</td>\n",
       "      <td>...</td>\n",
       "      <td>3.9099</td>\n",
       "      <td>3.3478</td>\n",
       "      <td>5.5201</td>\n",
       "      <td>5.6930</td>\n",
       "      <td>4.2072</td>\n",
       "      <td>4.2788</td>\n",
       "      <td>4.0790</td>\n",
       "      <td>4.7856</td>\n",
       "      <td>3.5387</td>\n",
       "      <td>4.7071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000278814.1</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 7430 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample  GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
       "0   ENSG00000242268.2                   -9.9658                   -9.9658   \n",
       "1   ENSG00000259041.1                   -9.9658                   -9.9658   \n",
       "2   ENSG00000270112.3                   -4.2934                    0.0014   \n",
       "3  ENSG00000167578.16                    5.1190                    4.1277   \n",
       "4   ENSG00000278814.1                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
       "0                   -9.9658                   -1.2481   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                   -9.9658                   -5.5735   \n",
       "3                    4.4067                    5.6860   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  \\\n",
       "0                   -3.8160                   -1.7809   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                    0.3573                   -9.9658   \n",
       "3                    4.0357                    4.6849   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-2226-SM-5N9CH  GTEX-1117F-2426-SM-5EGGH  \\\n",
       "0                   -9.9658                   -9.9658   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                   -6.5064                   -5.0116   \n",
       "3                    4.5009                    5.3954   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-2826-SM-5GZXL  ...  GTEX-ZZPU-0726-SM-5N9C8  \\\n",
       "0                   -3.6259  ...                  -0.8084   \n",
       "1                   -9.9658  ...                  -9.9658   \n",
       "2                   -9.9658  ...                  -4.0350   \n",
       "3                    4.9402  ...                   3.9099   \n",
       "4                   -9.9658  ...                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-0926-SM-5GZYT  GTEX-ZZPU-1226-SM-5N9CK  GTEX-ZZPU-1326-SM-5GZWS  \\\n",
       "0                  -9.9658                  -9.9658                  -9.9658   \n",
       "1                  -9.9658                  -9.9658                  -9.9658   \n",
       "2                  -1.8314                  -1.7809                  -2.0529   \n",
       "3                   3.3478                   5.5201                   5.6930   \n",
       "4                  -9.9658                  -9.9658                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-1426-SM-5GZZ6  GTEX-ZZPU-1826-SM-5E43L  GTEX-ZZPU-2126-SM-5EGIU  \\\n",
       "0                  -4.0350                  -9.9658                  -9.9658   \n",
       "1                  -9.9658                  -9.9658                  -9.9658   \n",
       "2                  -6.5064                  -3.6259                  -9.9658   \n",
       "3                   4.2072                   4.2788                   4.0790   \n",
       "4                  -9.9658                  -9.9658                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2626-SM-5E45Y  GTEX-ZZPU-2726-SM-5NQ8O  \n",
       "0                  -9.9658                  -9.9658                  -3.6259  \n",
       "1                  -9.9658                  -9.9658                  -9.9658  \n",
       "2                  -9.9658                  -2.8262                  -9.9658  \n",
       "3                   4.7856                   3.5387                   4.7071  \n",
       "4                  -9.9658                  -9.9658                  -9.9658  \n",
       "\n",
       "[5 rows x 7430 columns]"
      ]
     },
     "execution_count": 16,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gexpr_normal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>TCGA-02-0047-01</th>\n",
       "      <th>TCGA-02-0055-01</th>\n",
       "      <th>TCGA-02-2483-01</th>\n",
       "      <th>TCGA-02-2485-01</th>\n",
       "      <th>TCGA-04-1331-01</th>\n",
       "      <th>TCGA-04-1332-01</th>\n",
       "      <th>TCGA-04-1337-01</th>\n",
       "      <th>TCGA-04-1338-01</th>\n",
       "      <th>TCGA-04-1341-01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-ZP-A9D4-01</th>\n",
       "      <th>TCGA-ZQ-A9CR-01</th>\n",
       "      <th>TCGA-ZR-A9CJ-01</th>\n",
       "      <th>TCGA-ZS-A9CD-01</th>\n",
       "      <th>TCGA-ZS-A9CE-01</th>\n",
       "      <th>TCGA-ZS-A9CF-01</th>\n",
       "      <th>TCGA-ZS-A9CG-01</th>\n",
       "      <th>TCGA-ZT-A8OM-01</th>\n",
       "      <th>TCGA-ZU-A8S4-01</th>\n",
       "      <th>TCGA-ZX-AA5X-01</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000242268.2</td>\n",
       "      <td>0.0158</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-0.6193</td>\n",
       "      <td>-5.0116</td>\n",
       "      <td>-3.4580</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-4.6082</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000259041.1</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000270112.3</td>\n",
       "      <td>-0.8863</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>-5.5735</td>\n",
       "      <td>-1.5522</td>\n",
       "      <td>0.4761</td>\n",
       "      <td>-2.4659</td>\n",
       "      <td>-5.0116</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-2.7274</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-6.5064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000167578.16</td>\n",
       "      <td>5.9750</td>\n",
       "      <td>5.2506</td>\n",
       "      <td>5.4519</td>\n",
       "      <td>6.5390</td>\n",
       "      <td>4.9514</td>\n",
       "      <td>6.4481</td>\n",
       "      <td>5.1595</td>\n",
       "      <td>4.0506</td>\n",
       "      <td>4.5249</td>\n",
       "      <td>...</td>\n",
       "      <td>3.6136</td>\n",
       "      <td>4.4700</td>\n",
       "      <td>4.1780</td>\n",
       "      <td>4.5547</td>\n",
       "      <td>3.6737</td>\n",
       "      <td>4.9331</td>\n",
       "      <td>3.7646</td>\n",
       "      <td>5.5201</td>\n",
       "      <td>5.4216</td>\n",
       "      <td>4.7991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000278814.1</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 9186 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample  TCGA-02-0047-01  TCGA-02-0055-01  TCGA-02-2483-01  \\\n",
       "0   ENSG00000242268.2           0.0158          -9.9658          -9.9658   \n",
       "1   ENSG00000259041.1          -9.9658          -9.9658          -9.9658   \n",
       "2   ENSG00000270112.3          -0.8863          -3.8160          -5.5735   \n",
       "3  ENSG00000167578.16           5.9750           5.2506           5.4519   \n",
       "4   ENSG00000278814.1          -9.9658          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-02-2485-01  TCGA-04-1331-01  TCGA-04-1332-01  TCGA-04-1337-01  \\\n",
       "0          -0.6193          -5.0116          -3.4580          -3.8160   \n",
       "1          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "2          -1.5522           0.4761          -2.4659          -5.0116   \n",
       "3           6.5390           4.9514           6.4481           5.1595   \n",
       "4          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-04-1338-01  TCGA-04-1341-01  ...  TCGA-ZP-A9D4-01  TCGA-ZQ-A9CR-01  \\\n",
       "0          -9.9658          -9.9658  ...          -9.9658          -9.9658   \n",
       "1          -9.9658          -9.9658  ...          -9.9658          -9.9658   \n",
       "2          -4.0350          -2.7274  ...          -9.9658          -9.9658   \n",
       "3           4.0506           4.5249  ...           3.6136           4.4700   \n",
       "4          -9.9658          -9.9658  ...          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-ZR-A9CJ-01  TCGA-ZS-A9CD-01  TCGA-ZS-A9CE-01  TCGA-ZS-A9CF-01  \\\n",
       "0          -9.9658          -4.6082          -9.9658          -9.9658   \n",
       "1          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "2          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "3           4.1780           4.5547           3.6737           4.9331   \n",
       "4          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-ZS-A9CG-01  TCGA-ZT-A8OM-01  TCGA-ZU-A8S4-01  TCGA-ZX-AA5X-01  \n",
       "0          -9.9658          -3.6259          -9.9658          -9.9658  \n",
       "1          -9.9658          -9.9658          -9.9658          -9.9658  \n",
       "2          -9.9658          -9.9658          -9.9658          -6.5064  \n",
       "3           3.7646           5.5201           5.4216           4.7991  \n",
       "4          -9.9658          -9.9658          -9.9658          -9.9658  \n",
       "\n",
       "[5 rows x 9186 columns]"
      ]
     },
     "execution_count": 17,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gexpr_primary.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-1117F-0426-SM-5EGHI</th>\n",
       "      <th>GTEX-1117F-0526-SM-5EGHJ</th>\n",
       "      <th>GTEX-1117F-0626-SM-5N9CS</th>\n",
       "      <th>GTEX-1117F-0726-SM-5GIEN</th>\n",
       "      <th>GTEX-1117F-1326-SM-5EGHH</th>\n",
       "      <th>GTEX-1117F-2226-SM-5N9CH</th>\n",
       "      <th>GTEX-1117F-2426-SM-5EGGH</th>\n",
       "      <th>GTEX-1117F-2826-SM-5GZXL</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZZPU-0726-SM-5N9C8</th>\n",
       "      <th>GTEX-ZZPU-0926-SM-5GZYT</th>\n",
       "      <th>GTEX-ZZPU-1226-SM-5N9CK</th>\n",
       "      <th>GTEX-ZZPU-1326-SM-5GZWS</th>\n",
       "      <th>GTEX-ZZPU-1426-SM-5GZZ6</th>\n",
       "      <th>GTEX-ZZPU-1826-SM-5E43L</th>\n",
       "      <th>GTEX-ZZPU-2126-SM-5EGIU</th>\n",
       "      <th>GTEX-ZZPU-2226-SM-5EGIV</th>\n",
       "      <th>GTEX-ZZPU-2626-SM-5E45Y</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000242268.2</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-1.2481</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>-1.7809</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.8084</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000259041.1</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000270112.3</td>\n",
       "      <td>-4.2934</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-5.5735</td>\n",
       "      <td>0.3573</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-6.5064</td>\n",
       "      <td>-5.0116</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-1.8314</td>\n",
       "      <td>-1.7809</td>\n",
       "      <td>-2.0529</td>\n",
       "      <td>-6.5064</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-2.8262</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000167578.16</td>\n",
       "      <td>5.1190</td>\n",
       "      <td>4.1277</td>\n",
       "      <td>4.4067</td>\n",
       "      <td>5.6860</td>\n",
       "      <td>4.0357</td>\n",
       "      <td>4.6849</td>\n",
       "      <td>4.5009</td>\n",
       "      <td>5.3954</td>\n",
       "      <td>4.9402</td>\n",
       "      <td>...</td>\n",
       "      <td>3.9099</td>\n",
       "      <td>3.3478</td>\n",
       "      <td>5.5201</td>\n",
       "      <td>5.6930</td>\n",
       "      <td>4.2072</td>\n",
       "      <td>4.2788</td>\n",
       "      <td>4.0790</td>\n",
       "      <td>4.7856</td>\n",
       "      <td>3.5387</td>\n",
       "      <td>4.7071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000278814.1</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 7430 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "               sample  GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
       "0   ENSG00000242268.2                   -9.9658                   -9.9658   \n",
       "1   ENSG00000259041.1                   -9.9658                   -9.9658   \n",
       "2   ENSG00000270112.3                   -4.2934                    0.0014   \n",
       "3  ENSG00000167578.16                    5.1190                    4.1277   \n",
       "4   ENSG00000278814.1                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
       "0                   -9.9658                   -1.2481   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                   -9.9658                   -5.5735   \n",
       "3                    4.4067                    5.6860   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  \\\n",
       "0                   -3.8160                   -1.7809   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                    0.3573                   -9.9658   \n",
       "3                    4.0357                    4.6849   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-2226-SM-5N9CH  GTEX-1117F-2426-SM-5EGGH  \\\n",
       "0                   -9.9658                   -9.9658   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                   -6.5064                   -5.0116   \n",
       "3                    4.5009                    5.3954   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-2826-SM-5GZXL  ...  GTEX-ZZPU-0726-SM-5N9C8  \\\n",
       "0                   -3.6259  ...                  -0.8084   \n",
       "1                   -9.9658  ...                  -9.9658   \n",
       "2                   -9.9658  ...                  -4.0350   \n",
       "3                    4.9402  ...                   3.9099   \n",
       "4                   -9.9658  ...                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-0926-SM-5GZYT  GTEX-ZZPU-1226-SM-5N9CK  GTEX-ZZPU-1326-SM-5GZWS  \\\n",
       "0                  -9.9658                  -9.9658                  -9.9658   \n",
       "1                  -9.9658                  -9.9658                  -9.9658   \n",
       "2                  -1.8314                  -1.7809                  -2.0529   \n",
       "3                   3.3478                   5.5201                   5.6930   \n",
       "4                  -9.9658                  -9.9658                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-1426-SM-5GZZ6  GTEX-ZZPU-1826-SM-5E43L  GTEX-ZZPU-2126-SM-5EGIU  \\\n",
       "0                  -4.0350                  -9.9658                  -9.9658   \n",
       "1                  -9.9658                  -9.9658                  -9.9658   \n",
       "2                  -6.5064                  -3.6259                  -9.9658   \n",
       "3                   4.2072                   4.2788                   4.0790   \n",
       "4                  -9.9658                  -9.9658                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2626-SM-5E45Y  GTEX-ZZPU-2726-SM-5NQ8O  \n",
       "0                  -9.9658                  -9.9658                  -3.6259  \n",
       "1                  -9.9658                  -9.9658                  -9.9658  \n",
       "2                  -9.9658                  -2.8262                  -9.9658  \n",
       "3                   4.7856                   3.5387                   4.7071  \n",
       "4                  -9.9658                  -9.9658                  -9.9658  \n",
       "\n",
       "[5 rows x 7430 columns]"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gexpr_normal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {
    "id": "s_4JIRYMnuti"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "0\n",
      "before\n",
      "                sample  GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
      "28551  ENSG00000198712                   12.8384                   14.7048   \n",
      "\n",
      "       GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
      "28551                   12.8456                   12.6928   \n",
      "\n",
      "       GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  \\\n",
      "28551                   14.8495                   13.4601   \n",
      "\n",
      "       GTEX-1117F-2226-SM-5N9CH  GTEX-1117F-2426-SM-5EGGH  \\\n",
      "28551                   11.1638                   12.1732   \n",
      "\n",
      "       GTEX-1117F-2826-SM-5GZXL  ...  GTEX-ZZPU-0726-SM-5N9C8  \\\n",
      "28551                   13.7657  ...                  13.3606   \n",
      "\n",
      "       GTEX-ZZPU-0926-SM-5GZYT  GTEX-ZZPU-1226-SM-5N9CK  \\\n",
      "28551                  15.3773                  14.9873   \n",
      "\n",
      "       GTEX-ZZPU-1326-SM-5GZWS  GTEX-ZZPU-1426-SM-5GZZ6  \\\n",
      "28551                  14.6528                   14.776   \n",
      "\n",
      "       GTEX-ZZPU-1826-SM-5E43L  GTEX-ZZPU-2126-SM-5EGIU  \\\n",
      "28551                   14.834                  13.4387   \n",
      "\n",
      "       GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2626-SM-5E45Y  \\\n",
      "28551                  14.1687                  15.1614   \n",
      "\n",
      "       GTEX-ZZPU-2726-SM-5NQ8O  \n",
      "28551                  14.3728  \n",
      "\n",
      "[1 rows x 7430 columns]\n",
      "\n",
      "control merge is correct\n",
      "       ensembl_gene_id  GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
      "28551  ENSG00000198712                   12.8384                   14.7048   \n",
      "\n",
      "       GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
      "28551                   12.8456                   12.6928   \n",
      "\n",
      "       GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  \\\n",
      "28551                   14.8495                   13.4601   \n",
      "\n",
      "       GTEX-1117F-2226-SM-5N9CH  GTEX-1117F-2426-SM-5EGGH  \\\n",
      "28551                   11.1638                   12.1732   \n",
      "\n",
      "       GTEX-1117F-2826-SM-5GZXL  ...  GTEX-ZZPU-0726-SM-5N9C8  \\\n",
      "28551                   13.7657  ...                  13.3606   \n",
      "\n",
      "       GTEX-ZZPU-0926-SM-5GZYT  GTEX-ZZPU-1226-SM-5N9CK  \\\n",
      "28551                  15.3773                  14.9873   \n",
      "\n",
      "       GTEX-ZZPU-1326-SM-5GZWS  GTEX-ZZPU-1426-SM-5GZZ6  \\\n",
      "28551                  14.6528                   14.776   \n",
      "\n",
      "       GTEX-ZZPU-1826-SM-5E43L  GTEX-ZZPU-2126-SM-5EGIU  \\\n",
      "28551                   14.834                  13.4387   \n",
      "\n",
      "       GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2626-SM-5E45Y  \\\n",
      "28551                  14.1687                  15.1614   \n",
      "\n",
      "       GTEX-ZZPU-2726-SM-5NQ8O  \n",
      "28551                  14.3728  \n",
      "\n",
      "[1 rows x 7430 columns]\n",
      "0\n",
      "before\n",
      "                sample  TCGA-02-0047-01  TCGA-02-0055-01  TCGA-02-2483-01  \\\n",
      "28551  ENSG00000198712          13.5251          12.3893          13.4086   \n",
      "\n",
      "       TCGA-02-2485-01  TCGA-04-1331-01  TCGA-04-1332-01  TCGA-04-1337-01  \\\n",
      "28551          11.6139          14.2857          11.9431          14.3388   \n",
      "\n",
      "       TCGA-04-1338-01  TCGA-04-1341-01  ...  TCGA-ZP-A9D4-01  \\\n",
      "28551          14.1084          12.9949  ...          14.7449   \n",
      "\n",
      "       TCGA-ZQ-A9CR-01  TCGA-ZR-A9CJ-01  TCGA-ZS-A9CD-01  TCGA-ZS-A9CE-01  \\\n",
      "28551           14.156          13.2731           13.605          13.0069   \n",
      "\n",
      "       TCGA-ZS-A9CF-01  TCGA-ZS-A9CG-01  TCGA-ZT-A8OM-01  TCGA-ZU-A8S4-01  \\\n",
      "28551          13.9678          14.3166          11.0066          12.9042   \n",
      "\n",
      "       TCGA-ZX-AA5X-01  \n",
      "28551          14.5414  \n",
      "\n",
      "[1 rows x 9186 columns]\n",
      "\n",
      "control merge is correct\n",
      "       ensembl_gene_id  TCGA-02-0047-01  TCGA-02-0055-01  TCGA-02-2483-01  \\\n",
      "28551  ENSG00000198712          13.5251          12.3893          13.4086   \n",
      "\n",
      "       TCGA-02-2485-01  TCGA-04-1331-01  TCGA-04-1332-01  TCGA-04-1337-01  \\\n",
      "28551          11.6139          14.2857          11.9431          14.3388   \n",
      "\n",
      "       TCGA-04-1338-01  TCGA-04-1341-01  ...  TCGA-ZP-A9D4-01  \\\n",
      "28551          14.1084          12.9949  ...          14.7449   \n",
      "\n",
      "       TCGA-ZQ-A9CR-01  TCGA-ZR-A9CJ-01  TCGA-ZS-A9CD-01  TCGA-ZS-A9CE-01  \\\n",
      "28551           14.156          13.2731           13.605          13.0069   \n",
      "\n",
      "       TCGA-ZS-A9CF-01  TCGA-ZS-A9CG-01  TCGA-ZT-A8OM-01  TCGA-ZU-A8S4-01  \\\n",
      "28551          13.9678          14.3166          11.0066          12.9042   \n",
      "\n",
      "       TCGA-ZX-AA5X-01  \n",
      "28551          14.5414  \n",
      "\n",
      "[1 rows x 9186 columns]\n"
     ]
    }
   ],
   "source": [
    "#Preparation protein coding gene expression\n",
    "#remove dot in ENSEMBL name. Function split created at the beginning\n",
    "data = [df_gexpr_normal, df_gexpr_primary]\n",
    "\n",
    "df_protcoding_exp = []\n",
    "for df in data: \n",
    "  #remove dot in ens name\n",
    "  df[\"sample\"] = df[\"sample\"].apply(split)\n",
    "  #see if have duplicate\n",
    "  print(df.duplicated(subset=['sample']).sum())\n",
    "    \n",
    "  print('before')\n",
    "  print(df.loc[df['sample'] == 'ENSG00000198712'])\n",
    "  print('')\n",
    "\n",
    "  #Merge with prot coding list\n",
    "  df.rename(columns = {'sample' : 'ensembl_gene_id'}, inplace = True)\n",
    "  dfmerged = protein_coding_list.merge(df)\n",
    "  df_protcoding_exp.append(dfmerged)\n",
    "\n",
    "  #control merge is correct\n",
    "  print(\"control merge is correct\")\n",
    "  print(df.loc[df['ensembl_gene_id'] == 'ENSG00000198712'])\n",
    "  #print(df_protcoding_exp[0].loc[df_protcoding_exp[0]['ensembl_gene_id'] == 'ENSG00000198712'])\n",
    "\n",
    "normal_protcoding = df_protcoding_exp[0]\n",
    "primary_protcoding = df_protcoding_exp[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "normal_protcoding = df_protcoding_exp[0]\n",
    "primary_protcoding = df_protcoding_exp[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "ename": "TypeError",
     "evalue": "list indices must be integers or slices, not str",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-21-58e576f28e03>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mdf_protcoding_exp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0mdf_protcoding_exp\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m'ensembl_gene_id'\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m==\u001b[0m \u001b[0;34m'ENSG00000198712'\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;31mTypeError\u001b[0m: list indices must be integers or slices, not str"
     ]
    }
   ],
   "source": [
    "#df_protcoding_exp[df_protcoding_exp['ensembl_gene_id'] == 'ENSG00000198712']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "list"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "type(df_protcoding_exp)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ATG7_3</th>\n",
       "      <th>ATG7(total)</th>\n",
       "      <th>ATG7(1)</th>\n",
       "      <th>ATG7(2)</th>\n",
       "      <th>ATG7(3)</th>\n",
       "      <th>log2_+1_ATG7(total)</th>\n",
       "      <th>log2_+1_ATG7(1)</th>\n",
       "      <th>log2_+1_ATG7(2)</th>\n",
       "      <th>ATG7_prot_tot</th>\n",
       "      <th>ATG7_1%</th>\n",
       "      <th>ATG7_2%</th>\n",
       "      <th>ATG7_3%</th>\n",
       "      <th>ATG7(1)norm</th>\n",
       "      <th>ATG7(2)norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>656</th>\n",
       "      <td>GTEX-ZTTD-0326-SM-57WFW</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>5.042</td>\n",
       "      <td>3.860</td>\n",
       "      <td>2.390</td>\n",
       "      <td>0.2642</td>\n",
       "      <td>32.944283</td>\n",
       "      <td>14.519306</td>\n",
       "      <td>5.240574</td>\n",
       "      <td>1.199970e+00</td>\n",
       "      <td>5.085097</td>\n",
       "      <td>3.955992</td>\n",
       "      <td>2.641679</td>\n",
       "      <td>20.959850</td>\n",
       "      <td>69.271996</td>\n",
       "      <td>25.002916</td>\n",
       "      <td>5.725088</td>\n",
       "      <td>0.440723</td>\n",
       "      <td>0.159074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>657</th>\n",
       "      <td>GTEX-PX3G-1626-SM-2S1PT</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>4.252</td>\n",
       "      <td>2.322</td>\n",
       "      <td>1.956</td>\n",
       "      <td>1.1320</td>\n",
       "      <td>19.052710</td>\n",
       "      <td>4.999249</td>\n",
       "      <td>3.878848</td>\n",
       "      <td>2.190624e+00</td>\n",
       "      <td>4.325725</td>\n",
       "      <td>2.584782</td>\n",
       "      <td>2.286540</td>\n",
       "      <td>11.068720</td>\n",
       "      <td>45.165557</td>\n",
       "      <td>35.043325</td>\n",
       "      <td>19.791118</td>\n",
       "      <td>0.262390</td>\n",
       "      <td>0.203585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>658</th>\n",
       "      <td>GTEX-OXRO-1726-SM-3LK6C</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.909</td>\n",
       "      <td>2.602</td>\n",
       "      <td>1.345</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>15.020948</td>\n",
       "      <td>6.070277</td>\n",
       "      <td>2.539302</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>4.001888</td>\n",
       "      <td>2.821767</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>8.609579</td>\n",
       "      <td>70.506086</td>\n",
       "      <td>29.493916</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>0.404121</td>\n",
       "      <td>0.169051</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>659</th>\n",
       "      <td>GTEX-13JVG-0126-SM-5L3D3</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.854</td>\n",
       "      <td>2.568</td>\n",
       "      <td>1.428</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>14.459044</td>\n",
       "      <td>5.928868</td>\n",
       "      <td>2.689734</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>3.950379</td>\n",
       "      <td>2.792620</td>\n",
       "      <td>1.883517</td>\n",
       "      <td>8.618602</td>\n",
       "      <td>68.791526</td>\n",
       "      <td>31.208476</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>0.410046</td>\n",
       "      <td>0.186024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>660</th>\n",
       "      <td>GTEX-NPJ8-1626-SM-2HMIY</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.822</td>\n",
       "      <td>2.164</td>\n",
       "      <td>1.293</td>\n",
       "      <td>-1.5100</td>\n",
       "      <td>14.141840</td>\n",
       "      <td>4.480557</td>\n",
       "      <td>2.449371</td>\n",
       "      <td>3.501112e-01</td>\n",
       "      <td>3.920469</td>\n",
       "      <td>2.454322</td>\n",
       "      <td>1.786333</td>\n",
       "      <td>7.280039</td>\n",
       "      <td>61.545783</td>\n",
       "      <td>33.645022</td>\n",
       "      <td>4.809194</td>\n",
       "      <td>0.316830</td>\n",
       "      <td>0.173200</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                       sample    Sample_Type Primary_Site   ATG7  ATG7_1  \\\n",
       "656   GTEX-ZTTD-0326-SM-57WFW  Normal Tissue       Muscle  5.042   3.860   \n",
       "657   GTEX-PX3G-1626-SM-2S1PT  Normal Tissue       Muscle  4.252   2.322   \n",
       "658   GTEX-OXRO-1726-SM-3LK6C  Normal Tissue       Muscle  3.909   2.602   \n",
       "659  GTEX-13JVG-0126-SM-5L3D3  Normal Tissue       Muscle  3.854   2.568   \n",
       "660   GTEX-NPJ8-1626-SM-2HMIY  Normal Tissue       Muscle  3.822   2.164   \n",
       "\n",
       "     ATG7_2  ATG7_3  ATG7(total)    ATG7(1)   ATG7(2)       ATG7(3)  \\\n",
       "656   2.390  0.2642    32.944283  14.519306  5.240574  1.199970e+00   \n",
       "657   1.956  1.1320    19.052710   4.999249  3.878848  2.190624e+00   \n",
       "658   1.345 -9.9660    15.020948   6.070277  2.539302 -1.495113e-07   \n",
       "659   1.428 -9.9660    14.459044   5.928868  2.689734 -1.495113e-07   \n",
       "660   1.293 -1.5100    14.141840   4.480557  2.449371  3.501112e-01   \n",
       "\n",
       "     log2_+1_ATG7(total)  log2_+1_ATG7(1)  log2_+1_ATG7(2)  ATG7_prot_tot  \\\n",
       "656             5.085097         3.955992         2.641679      20.959850   \n",
       "657             4.325725         2.584782         2.286540      11.068720   \n",
       "658             4.001888         2.821767         1.823465       8.609579   \n",
       "659             3.950379         2.792620         1.883517       8.618602   \n",
       "660             3.920469         2.454322         1.786333       7.280039   \n",
       "\n",
       "       ATG7_1%    ATG7_2%    ATG7_3%  ATG7(1)norm  ATG7(2)norm  \n",
       "656  69.271996  25.002916   5.725088     0.440723     0.159074  \n",
       "657  45.165557  35.043325  19.791118     0.262390     0.203585  \n",
       "658  70.506086  29.493916  -0.000002     0.404121     0.169051  \n",
       "659  68.791526  31.208476  -0.000002     0.410046     0.186024  \n",
       "660  61.545783  33.645022   4.809194     0.316830     0.173200  "
      ]
     },
     "execution_count": 25,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_ATG7_Normal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>entrezgene_id</th>\n",
       "      <th>transcript_biotype</th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-1117F-0426-SM-5EGHI</th>\n",
       "      <th>GTEX-1117F-0526-SM-5EGHJ</th>\n",
       "      <th>GTEX-1117F-0626-SM-5N9CS</th>\n",
       "      <th>GTEX-1117F-0726-SM-5GIEN</th>\n",
       "      <th>GTEX-1117F-1326-SM-5EGHH</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZZPU-0726-SM-5N9C8</th>\n",
       "      <th>GTEX-ZZPU-0926-SM-5GZYT</th>\n",
       "      <th>GTEX-ZZPU-1226-SM-5N9CK</th>\n",
       "      <th>GTEX-ZZPU-1326-SM-5GZWS</th>\n",
       "      <th>GTEX-ZZPU-1426-SM-5GZZ6</th>\n",
       "      <th>GTEX-ZZPU-1826-SM-5E43L</th>\n",
       "      <th>GTEX-ZZPU-2126-SM-5EGIU</th>\n",
       "      <th>GTEX-ZZPU-2226-SM-5EGIV</th>\n",
       "      <th>GTEX-ZZPU-2626-SM-5E45Y</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>4535.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13.1350</td>\n",
       "      <td>14.6712</td>\n",
       "      <td>12.8924</td>\n",
       "      <td>12.6263</td>\n",
       "      <td>13.8339</td>\n",
       "      <td>13.7962</td>\n",
       "      <td>...</td>\n",
       "      <td>12.8565</td>\n",
       "      <td>14.8073</td>\n",
       "      <td>14.3240</td>\n",
       "      <td>14.0776</td>\n",
       "      <td>14.1602</td>\n",
       "      <td>13.7105</td>\n",
       "      <td>12.8758</td>\n",
       "      <td>12.5436</td>\n",
       "      <td>14.6088</td>\n",
       "      <td>14.3228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>4536.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12.6215</td>\n",
       "      <td>14.2931</td>\n",
       "      <td>12.6612</td>\n",
       "      <td>13.0833</td>\n",
       "      <td>14.1782</td>\n",
       "      <td>13.8157</td>\n",
       "      <td>...</td>\n",
       "      <td>13.0353</td>\n",
       "      <td>14.6549</td>\n",
       "      <td>14.3492</td>\n",
       "      <td>14.2711</td>\n",
       "      <td>14.3810</td>\n",
       "      <td>14.2432</td>\n",
       "      <td>13.3940</td>\n",
       "      <td>12.8245</td>\n",
       "      <td>14.2491</td>\n",
       "      <td>14.3830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>4512.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12.6246</td>\n",
       "      <td>14.5995</td>\n",
       "      <td>12.5008</td>\n",
       "      <td>11.3334</td>\n",
       "      <td>15.3590</td>\n",
       "      <td>12.2692</td>\n",
       "      <td>...</td>\n",
       "      <td>13.3965</td>\n",
       "      <td>15.9901</td>\n",
       "      <td>15.2750</td>\n",
       "      <td>15.1726</td>\n",
       "      <td>14.8551</td>\n",
       "      <td>14.6084</td>\n",
       "      <td>13.5241</td>\n",
       "      <td>14.0931</td>\n",
       "      <td>15.5290</td>\n",
       "      <td>14.3970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>4513.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12.8384</td>\n",
       "      <td>14.7048</td>\n",
       "      <td>12.8456</td>\n",
       "      <td>12.6928</td>\n",
       "      <td>14.8495</td>\n",
       "      <td>13.4601</td>\n",
       "      <td>...</td>\n",
       "      <td>13.3606</td>\n",
       "      <td>15.3773</td>\n",
       "      <td>14.9873</td>\n",
       "      <td>14.6528</td>\n",
       "      <td>14.7760</td>\n",
       "      <td>14.8340</td>\n",
       "      <td>13.4387</td>\n",
       "      <td>14.1687</td>\n",
       "      <td>15.1614</td>\n",
       "      <td>14.3728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>4509.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13.9763</td>\n",
       "      <td>15.7851</td>\n",
       "      <td>13.9252</td>\n",
       "      <td>14.1037</td>\n",
       "      <td>16.3241</td>\n",
       "      <td>15.3932</td>\n",
       "      <td>...</td>\n",
       "      <td>14.3118</td>\n",
       "      <td>16.4238</td>\n",
       "      <td>15.5251</td>\n",
       "      <td>15.8844</td>\n",
       "      <td>15.8104</td>\n",
       "      <td>15.4948</td>\n",
       "      <td>14.9172</td>\n",
       "      <td>14.8163</td>\n",
       "      <td>16.0166</td>\n",
       "      <td>15.2887</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 7433 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ensembl_gene_id hgnc_symbol  entrezgene_id transcript_biotype  \\\n",
       "0  ENSG00000198888      MT-ND1         4535.0     protein_coding   \n",
       "1  ENSG00000198763      MT-ND2         4536.0     protein_coding   \n",
       "2  ENSG00000198804      MT-CO1         4512.0     protein_coding   \n",
       "3  ENSG00000198712      MT-CO2         4513.0     protein_coding   \n",
       "4  ENSG00000228253     MT-ATP8         4509.0     protein_coding   \n",
       "\n",
       "   GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
       "0                   13.1350                   14.6712   \n",
       "1                   12.6215                   14.2931   \n",
       "2                   12.6246                   14.5995   \n",
       "3                   12.8384                   14.7048   \n",
       "4                   13.9763                   15.7851   \n",
       "\n",
       "   GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
       "0                   12.8924                   12.6263   \n",
       "1                   12.6612                   13.0833   \n",
       "2                   12.5008                   11.3334   \n",
       "3                   12.8456                   12.6928   \n",
       "4                   13.9252                   14.1037   \n",
       "\n",
       "   GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  ...  \\\n",
       "0                   13.8339                   13.7962  ...   \n",
       "1                   14.1782                   13.8157  ...   \n",
       "2                   15.3590                   12.2692  ...   \n",
       "3                   14.8495                   13.4601  ...   \n",
       "4                   16.3241                   15.3932  ...   \n",
       "\n",
       "   GTEX-ZZPU-0726-SM-5N9C8  GTEX-ZZPU-0926-SM-5GZYT  GTEX-ZZPU-1226-SM-5N9CK  \\\n",
       "0                  12.8565                  14.8073                  14.3240   \n",
       "1                  13.0353                  14.6549                  14.3492   \n",
       "2                  13.3965                  15.9901                  15.2750   \n",
       "3                  13.3606                  15.3773                  14.9873   \n",
       "4                  14.3118                  16.4238                  15.5251   \n",
       "\n",
       "   GTEX-ZZPU-1326-SM-5GZWS  GTEX-ZZPU-1426-SM-5GZZ6  GTEX-ZZPU-1826-SM-5E43L  \\\n",
       "0                  14.0776                  14.1602                  13.7105   \n",
       "1                  14.2711                  14.3810                  14.2432   \n",
       "2                  15.1726                  14.8551                  14.6084   \n",
       "3                  14.6528                  14.7760                  14.8340   \n",
       "4                  15.8844                  15.8104                  15.4948   \n",
       "\n",
       "   GTEX-ZZPU-2126-SM-5EGIU  GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2626-SM-5E45Y  \\\n",
       "0                  12.8758                  12.5436                  14.6088   \n",
       "1                  13.3940                  12.8245                  14.2491   \n",
       "2                  13.5241                  14.0931                  15.5290   \n",
       "3                  13.4387                  14.1687                  15.1614   \n",
       "4                  14.9172                  14.8163                  16.0166   \n",
       "\n",
       "   GTEX-ZZPU-2726-SM-5NQ8O  \n",
       "0                  14.3228  \n",
       "1                  14.3830  \n",
       "2                  14.3970  \n",
       "3                  14.3728  \n",
       "4                  15.2887  \n",
       "\n",
       "[5 rows x 7433 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_protcoding.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>GTEX-ZTTD-0326-SM-57WFW</th>\n",
       "      <th>GTEX-PX3G-1626-SM-2S1PT</th>\n",
       "      <th>GTEX-OXRO-1726-SM-3LK6C</th>\n",
       "      <th>GTEX-13JVG-0126-SM-5L3D3</th>\n",
       "      <th>GTEX-NPJ8-1626-SM-2HMIY</th>\n",
       "      <th>GTEX-ZC5H-0326-SM-5HL9F</th>\n",
       "      <th>GTEX-ZYY3-0526-SM-5E45G</th>\n",
       "      <th>GTEX-11TTK-0326-SM-5BC5N</th>\n",
       "      <th>GTEX-XV7Q-2926-SM-4BRUL</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZAJG-2826-SM-5HL8D</th>\n",
       "      <th>GTEX-1399U-1326-SM-5IJET</th>\n",
       "      <th>GTEX-132AR-1526-SM-5KM1L</th>\n",
       "      <th>GTEX-139D8-2526-SM-5N9G3</th>\n",
       "      <th>GTEX-ZV6S-1526-SM-5NQ7O</th>\n",
       "      <th>GTEX-WWYW-2826-SM-4MVOQ</th>\n",
       "      <th>GTEX-YFC4-0126-101855-SM-5CVLZ</th>\n",
       "      <th>GTEX-YEC4-1026-101814-SM-5P9FX</th>\n",
       "      <th>GTEX-YEC3-2526-101809-SM-5CVLT</th>\n",
       "      <th>GTEX-YEC3-1426-101806-SM-5PNXX</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ATG7_1</td>\n",
       "      <td>3.86</td>\n",
       "      <td>2.322</td>\n",
       "      <td>2.602</td>\n",
       "      <td>2.568</td>\n",
       "      <td>2.164</td>\n",
       "      <td>2.039</td>\n",
       "      <td>1.619</td>\n",
       "      <td>2.269</td>\n",
       "      <td>2.511</td>\n",
       "      <td>...</td>\n",
       "      <td>2.759</td>\n",
       "      <td>2.964</td>\n",
       "      <td>2.628</td>\n",
       "      <td>2.614</td>\n",
       "      <td>2.791</td>\n",
       "      <td>2.566</td>\n",
       "      <td>2.888</td>\n",
       "      <td>2.275</td>\n",
       "      <td>1.683</td>\n",
       "      <td>2.359</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>ATG7_2</td>\n",
       "      <td>2.39</td>\n",
       "      <td>1.956</td>\n",
       "      <td>1.345</td>\n",
       "      <td>1.428</td>\n",
       "      <td>1.293</td>\n",
       "      <td>1.008</td>\n",
       "      <td>1.532</td>\n",
       "      <td>1.967</td>\n",
       "      <td>1.215</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-0.9132</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-3.047</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>1.812</td>\n",
       "      <td>1.745</td>\n",
       "      <td>1.872</td>\n",
       "      <td>-0.3752</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>ATG7_3</td>\n",
       "      <td>0.2642</td>\n",
       "      <td>1.132</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-1.51</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-0.3752</td>\n",
       "      <td>-1.685</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-3.458</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>-9.966</td>\n",
       "      <td>0.2029</td>\n",
       "      <td>-0.1993</td>\n",
       "      <td>-0.4325</td>\n",
       "      <td>-9.966</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>ATG7(total)</td>\n",
       "      <td>32.944283</td>\n",
       "      <td>19.05271</td>\n",
       "      <td>15.020948</td>\n",
       "      <td>14.459044</td>\n",
       "      <td>14.14184</td>\n",
       "      <td>12.517589</td>\n",
       "      <td>12.091159</td>\n",
       "      <td>12.040974</td>\n",
       "      <td>11.711686</td>\n",
       "      <td>...</td>\n",
       "      <td>10.259293</td>\n",
       "      <td>10.202555</td>\n",
       "      <td>9.868657</td>\n",
       "      <td>9.278197</td>\n",
       "      <td>9.018203</td>\n",
       "      <td>8.729111</td>\n",
       "      <td>22.176118</td>\n",
       "      <td>13.094507</td>\n",
       "      <td>12.200613</td>\n",
       "      <td>8.710976</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>ATG7(1)</td>\n",
       "      <td>14.519306</td>\n",
       "      <td>4.999249</td>\n",
       "      <td>6.070277</td>\n",
       "      <td>5.928868</td>\n",
       "      <td>4.480557</td>\n",
       "      <td>4.108606</td>\n",
       "      <td>3.070621</td>\n",
       "      <td>4.818889</td>\n",
       "      <td>5.69915</td>\n",
       "      <td>...</td>\n",
       "      <td>6.768269</td>\n",
       "      <td>7.801844</td>\n",
       "      <td>6.180684</td>\n",
       "      <td>6.120987</td>\n",
       "      <td>6.920094</td>\n",
       "      <td>5.920653</td>\n",
       "      <td>7.401435</td>\n",
       "      <td>4.838976</td>\n",
       "      <td>3.20995</td>\n",
       "      <td>5.129146</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>ATG7(2)</td>\n",
       "      <td>5.240574</td>\n",
       "      <td>3.878848</td>\n",
       "      <td>2.539302</td>\n",
       "      <td>2.689734</td>\n",
       "      <td>2.449371</td>\n",
       "      <td>2.010121</td>\n",
       "      <td>2.890865</td>\n",
       "      <td>3.908543</td>\n",
       "      <td>2.320408</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.530006</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.119993</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>3.510287</td>\n",
       "      <td>3.350949</td>\n",
       "      <td>3.659397</td>\n",
       "      <td>0.769999</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>ATG7(3)</td>\n",
       "      <td>1.19997</td>\n",
       "      <td>2.190624</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.350111</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.769999</td>\n",
       "      <td>0.310003</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.089999</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>1.15001</td>\n",
       "      <td>0.869973</td>\n",
       "      <td>0.739977</td>\n",
       "      <td>-0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>log2_+1_ATG7(total)</td>\n",
       "      <td>5.085097</td>\n",
       "      <td>4.325725</td>\n",
       "      <td>4.001888</td>\n",
       "      <td>3.950379</td>\n",
       "      <td>3.920469</td>\n",
       "      <td>3.756766</td>\n",
       "      <td>3.710521</td>\n",
       "      <td>3.70498</td>\n",
       "      <td>3.668083</td>\n",
       "      <td>...</td>\n",
       "      <td>3.493044</td>\n",
       "      <td>3.485756</td>\n",
       "      <td>3.442102</td>\n",
       "      <td>3.361515</td>\n",
       "      <td>3.324552</td>\n",
       "      <td>3.282308</td>\n",
       "      <td>4.534567</td>\n",
       "      <td>3.817061</td>\n",
       "      <td>3.722533</td>\n",
       "      <td>3.279616</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>log2_+1_ATG7(1)</td>\n",
       "      <td>3.955992</td>\n",
       "      <td>2.584782</td>\n",
       "      <td>2.821767</td>\n",
       "      <td>2.79262</td>\n",
       "      <td>2.454322</td>\n",
       "      <td>2.35293</td>\n",
       "      <td>2.025249</td>\n",
       "      <td>2.540744</td>\n",
       "      <td>2.743978</td>\n",
       "      <td>...</td>\n",
       "      <td>2.957593</td>\n",
       "      <td>3.137806</td>\n",
       "      <td>2.844121</td>\n",
       "      <td>2.832077</td>\n",
       "      <td>2.985517</td>\n",
       "      <td>2.790908</td>\n",
       "      <td>3.070636</td>\n",
       "      <td>2.545715</td>\n",
       "      <td>2.073803</td>\n",
       "      <td>2.615686</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>log2_+1_ATG7(2)</td>\n",
       "      <td>2.641679</td>\n",
       "      <td>2.28654</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>1.883517</td>\n",
       "      <td>1.786333</td>\n",
       "      <td>1.589822</td>\n",
       "      <td>1.960091</td>\n",
       "      <td>2.295295</td>\n",
       "      <td>1.73136</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.613537</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.16349</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>2.173219</td>\n",
       "      <td>2.12133</td>\n",
       "      <td>2.220143</td>\n",
       "      <td>0.823748</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>ATG7_prot_tot</td>\n",
       "      <td>20.95985</td>\n",
       "      <td>11.06872</td>\n",
       "      <td>8.609579</td>\n",
       "      <td>8.618602</td>\n",
       "      <td>7.280039</td>\n",
       "      <td>6.118727</td>\n",
       "      <td>6.731484</td>\n",
       "      <td>9.037435</td>\n",
       "      <td>8.019558</td>\n",
       "      <td>...</td>\n",
       "      <td>6.768268</td>\n",
       "      <td>8.421849</td>\n",
       "      <td>6.180684</td>\n",
       "      <td>6.120987</td>\n",
       "      <td>7.040087</td>\n",
       "      <td>5.920653</td>\n",
       "      <td>12.061732</td>\n",
       "      <td>9.059898</td>\n",
       "      <td>7.609323</td>\n",
       "      <td>5.899145</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>ATG7_1%</td>\n",
       "      <td>69.271996</td>\n",
       "      <td>45.165557</td>\n",
       "      <td>70.506086</td>\n",
       "      <td>68.791526</td>\n",
       "      <td>61.545783</td>\n",
       "      <td>67.148051</td>\n",
       "      <td>45.615806</td>\n",
       "      <td>53.321425</td>\n",
       "      <td>71.065642</td>\n",
       "      <td>...</td>\n",
       "      <td>100.000004</td>\n",
       "      <td>92.638133</td>\n",
       "      <td>100.000005</td>\n",
       "      <td>100.000005</td>\n",
       "      <td>98.295572</td>\n",
       "      <td>100.000005</td>\n",
       "      <td>61.362955</td>\n",
       "      <td>53.410937</td>\n",
       "      <td>42.184431</td>\n",
       "      <td>86.947288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>ATG7_2%</td>\n",
       "      <td>25.002916</td>\n",
       "      <td>35.043325</td>\n",
       "      <td>29.493916</td>\n",
       "      <td>31.208476</td>\n",
       "      <td>33.645022</td>\n",
       "      <td>32.851952</td>\n",
       "      <td>42.94543</td>\n",
       "      <td>43.248366</td>\n",
       "      <td>28.93436</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>6.293226</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>1.70443</td>\n",
       "      <td>-0.000003</td>\n",
       "      <td>29.102679</td>\n",
       "      <td>36.986604</td>\n",
       "      <td>48.090963</td>\n",
       "      <td>13.052714</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>ATG7_3%</td>\n",
       "      <td>5.725088</td>\n",
       "      <td>19.791118</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>4.809194</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>11.438764</td>\n",
       "      <td>3.430209</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>1.068641</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>-0.000003</td>\n",
       "      <td>9.534366</td>\n",
       "      <td>9.60246</td>\n",
       "      <td>9.724606</td>\n",
       "      <td>-0.000003</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>ATG7(1)norm</td>\n",
       "      <td>0.440723</td>\n",
       "      <td>0.26239</td>\n",
       "      <td>0.404121</td>\n",
       "      <td>0.410046</td>\n",
       "      <td>0.31683</td>\n",
       "      <td>0.328227</td>\n",
       "      <td>0.253956</td>\n",
       "      <td>0.400208</td>\n",
       "      <td>0.486621</td>\n",
       "      <td>...</td>\n",
       "      <td>0.659721</td>\n",
       "      <td>0.764695</td>\n",
       "      <td>0.626294</td>\n",
       "      <td>0.659717</td>\n",
       "      <td>0.767347</td>\n",
       "      <td>0.678265</td>\n",
       "      <td>0.333757</td>\n",
       "      <td>0.369542</td>\n",
       "      <td>0.263097</td>\n",
       "      <td>0.588814</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>ATG7(2)norm</td>\n",
       "      <td>0.159074</td>\n",
       "      <td>0.203585</td>\n",
       "      <td>0.169051</td>\n",
       "      <td>0.186024</td>\n",
       "      <td>0.1732</td>\n",
       "      <td>0.160584</td>\n",
       "      <td>0.239089</td>\n",
       "      <td>0.324604</td>\n",
       "      <td>0.198128</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.051948</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.013306</td>\n",
       "      <td>-0.0</td>\n",
       "      <td>0.158291</td>\n",
       "      <td>0.255905</td>\n",
       "      <td>0.299935</td>\n",
       "      <td>0.088394</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>16 rows Ã— 7430 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                 sample GTEX-ZTTD-0326-SM-57WFW GTEX-PX3G-1626-SM-2S1PT  \\\n",
       "4                ATG7_1                    3.86                   2.322   \n",
       "5                ATG7_2                    2.39                   1.956   \n",
       "6                ATG7_3                  0.2642                   1.132   \n",
       "7           ATG7(total)               32.944283                19.05271   \n",
       "8               ATG7(1)               14.519306                4.999249   \n",
       "9               ATG7(2)                5.240574                3.878848   \n",
       "10              ATG7(3)                 1.19997                2.190624   \n",
       "11  log2_+1_ATG7(total)                5.085097                4.325725   \n",
       "12      log2_+1_ATG7(1)                3.955992                2.584782   \n",
       "13      log2_+1_ATG7(2)                2.641679                 2.28654   \n",
       "14        ATG7_prot_tot                20.95985                11.06872   \n",
       "15              ATG7_1%               69.271996               45.165557   \n",
       "16              ATG7_2%               25.002916               35.043325   \n",
       "17              ATG7_3%                5.725088               19.791118   \n",
       "18          ATG7(1)norm                0.440723                 0.26239   \n",
       "19          ATG7(2)norm                0.159074                0.203585   \n",
       "\n",
       "   GTEX-OXRO-1726-SM-3LK6C GTEX-13JVG-0126-SM-5L3D3 GTEX-NPJ8-1626-SM-2HMIY  \\\n",
       "4                    2.602                    2.568                   2.164   \n",
       "5                    1.345                    1.428                   1.293   \n",
       "6                   -9.966                   -9.966                   -1.51   \n",
       "7                15.020948                14.459044                14.14184   \n",
       "8                 6.070277                 5.928868                4.480557   \n",
       "9                 2.539302                 2.689734                2.449371   \n",
       "10                    -0.0                     -0.0                0.350111   \n",
       "11                4.001888                 3.950379                3.920469   \n",
       "12                2.821767                  2.79262                2.454322   \n",
       "13                1.823465                 1.883517                1.786333   \n",
       "14                8.609579                 8.618602                7.280039   \n",
       "15               70.506086                68.791526               61.545783   \n",
       "16               29.493916                31.208476               33.645022   \n",
       "17               -0.000002                -0.000002                4.809194   \n",
       "18                0.404121                 0.410046                 0.31683   \n",
       "19                0.169051                 0.186024                  0.1732   \n",
       "\n",
       "   GTEX-ZC5H-0326-SM-5HL9F GTEX-ZYY3-0526-SM-5E45G GTEX-11TTK-0326-SM-5BC5N  \\\n",
       "4                    2.039                   1.619                    2.269   \n",
       "5                    1.008                   1.532                    1.967   \n",
       "6                   -9.966                 -0.3752                   -1.685   \n",
       "7                12.517589               12.091159                12.040974   \n",
       "8                 4.108606                3.070621                 4.818889   \n",
       "9                 2.010121                2.890865                 3.908543   \n",
       "10                    -0.0                0.769999                 0.310003   \n",
       "11                3.756766                3.710521                  3.70498   \n",
       "12                 2.35293                2.025249                 2.540744   \n",
       "13                1.589822                1.960091                 2.295295   \n",
       "14                6.118727                6.731484                 9.037435   \n",
       "15               67.148051               45.615806                53.321425   \n",
       "16               32.851952                42.94543                43.248366   \n",
       "17               -0.000002               11.438764                 3.430209   \n",
       "18                0.328227                0.253956                 0.400208   \n",
       "19                0.160584                0.239089                 0.324604   \n",
       "\n",
       "   GTEX-XV7Q-2926-SM-4BRUL  ... GTEX-ZAJG-2826-SM-5HL8D  \\\n",
       "4                    2.511  ...                   2.759   \n",
       "5                    1.215  ...                  -9.966   \n",
       "6                   -9.966  ...                  -9.966   \n",
       "7                11.711686  ...               10.259293   \n",
       "8                  5.69915  ...                6.768269   \n",
       "9                 2.320408  ...                    -0.0   \n",
       "10                    -0.0  ...                    -0.0   \n",
       "11                3.668083  ...                3.493044   \n",
       "12                2.743978  ...                2.957593   \n",
       "13                 1.73136  ...                    -0.0   \n",
       "14                8.019558  ...                6.768268   \n",
       "15               71.065642  ...              100.000004   \n",
       "16                28.93436  ...               -0.000002   \n",
       "17               -0.000002  ...               -0.000002   \n",
       "18                0.486621  ...                0.659721   \n",
       "19                0.198128  ...                    -0.0   \n",
       "\n",
       "   GTEX-1399U-1326-SM-5IJET GTEX-132AR-1526-SM-5KM1L GTEX-139D8-2526-SM-5N9G3  \\\n",
       "4                     2.964                    2.628                    2.614   \n",
       "5                   -0.9132                   -9.966                   -9.966   \n",
       "6                    -3.458                   -9.966                   -9.966   \n",
       "7                 10.202555                 9.868657                 9.278197   \n",
       "8                  7.801844                 6.180684                 6.120987   \n",
       "9                  0.530006                     -0.0                     -0.0   \n",
       "10                 0.089999                     -0.0                     -0.0   \n",
       "11                 3.485756                 3.442102                 3.361515   \n",
       "12                 3.137806                 2.844121                 2.832077   \n",
       "13                 0.613537                     -0.0                     -0.0   \n",
       "14                 8.421849                 6.180684                 6.120987   \n",
       "15                92.638133               100.000005               100.000005   \n",
       "16                 6.293226                -0.000002                -0.000002   \n",
       "17                 1.068641                -0.000002                -0.000002   \n",
       "18                 0.764695                 0.626294                 0.659717   \n",
       "19                 0.051948                     -0.0                     -0.0   \n",
       "\n",
       "   GTEX-ZV6S-1526-SM-5NQ7O GTEX-WWYW-2826-SM-4MVOQ  \\\n",
       "4                    2.791                   2.566   \n",
       "5                   -3.047                  -9.966   \n",
       "6                   -9.966                  -9.966   \n",
       "7                 9.018203                8.729111   \n",
       "8                 6.920094                5.920653   \n",
       "9                 0.119993                    -0.0   \n",
       "10                    -0.0                    -0.0   \n",
       "11                3.324552                3.282308   \n",
       "12                2.985517                2.790908   \n",
       "13                 0.16349                    -0.0   \n",
       "14                7.040087                5.920653   \n",
       "15               98.295572              100.000005   \n",
       "16                 1.70443               -0.000003   \n",
       "17               -0.000002               -0.000003   \n",
       "18                0.767347                0.678265   \n",
       "19                0.013306                    -0.0   \n",
       "\n",
       "   GTEX-YFC4-0126-101855-SM-5CVLZ GTEX-YEC4-1026-101814-SM-5P9FX  \\\n",
       "4                           2.888                          2.275   \n",
       "5                           1.812                          1.745   \n",
       "6                          0.2029                        -0.1993   \n",
       "7                       22.176118                      13.094507   \n",
       "8                        7.401435                       4.838976   \n",
       "9                        3.510287                       3.350949   \n",
       "10                        1.15001                       0.869973   \n",
       "11                       4.534567                       3.817061   \n",
       "12                       3.070636                       2.545715   \n",
       "13                       2.173219                        2.12133   \n",
       "14                      12.061732                       9.059898   \n",
       "15                      61.362955                      53.410937   \n",
       "16                      29.102679                      36.986604   \n",
       "17                       9.534366                        9.60246   \n",
       "18                       0.333757                       0.369542   \n",
       "19                       0.158291                       0.255905   \n",
       "\n",
       "   GTEX-YEC3-2526-101809-SM-5CVLT GTEX-YEC3-1426-101806-SM-5PNXX  \n",
       "4                           1.683                          2.359  \n",
       "5                           1.872                        -0.3752  \n",
       "6                         -0.4325                         -9.966  \n",
       "7                       12.200613                       8.710976  \n",
       "8                         3.20995                       5.129146  \n",
       "9                        3.659397                       0.769999  \n",
       "10                       0.739977                           -0.0  \n",
       "11                       3.722533                       3.279616  \n",
       "12                       2.073803                       2.615686  \n",
       "13                       2.220143                       0.823748  \n",
       "14                       7.609323                       5.899145  \n",
       "15                      42.184431                      86.947288  \n",
       "16                      48.090963                      13.052714  \n",
       "17                       9.724606                      -0.000003  \n",
       "18                       0.263097                       0.588814  \n",
       "19                       0.299935                       0.088394  \n",
       "\n",
       "[16 rows x 7430 columns]"
      ]
     },
     "execution_count": 33,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "curr_df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>entrezgene_id</th>\n",
       "      <th>transcript_biotype</th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-1117F-0426-SM-5EGHI</th>\n",
       "      <th>GTEX-1117F-0526-SM-5EGHJ</th>\n",
       "      <th>GTEX-1117F-0626-SM-5N9CS</th>\n",
       "      <th>GTEX-1117F-0726-SM-5GIEN</th>\n",
       "      <th>GTEX-1117F-1326-SM-5EGHH</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZZPU-0726-SM-5N9C8</th>\n",
       "      <th>GTEX-ZZPU-0926-SM-5GZYT</th>\n",
       "      <th>GTEX-ZZPU-1226-SM-5N9CK</th>\n",
       "      <th>GTEX-ZZPU-1326-SM-5GZWS</th>\n",
       "      <th>GTEX-ZZPU-1426-SM-5GZZ6</th>\n",
       "      <th>GTEX-ZZPU-1826-SM-5E43L</th>\n",
       "      <th>GTEX-ZZPU-2126-SM-5EGIU</th>\n",
       "      <th>GTEX-ZZPU-2226-SM-5EGIV</th>\n",
       "      <th>GTEX-ZZPU-2626-SM-5E45Y</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>4535.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13.1350</td>\n",
       "      <td>14.6712</td>\n",
       "      <td>12.8924</td>\n",
       "      <td>12.6263</td>\n",
       "      <td>13.8339</td>\n",
       "      <td>13.7962</td>\n",
       "      <td>...</td>\n",
       "      <td>12.8565</td>\n",
       "      <td>14.8073</td>\n",
       "      <td>14.3240</td>\n",
       "      <td>14.0776</td>\n",
       "      <td>14.1602</td>\n",
       "      <td>13.7105</td>\n",
       "      <td>12.8758</td>\n",
       "      <td>12.5436</td>\n",
       "      <td>14.6088</td>\n",
       "      <td>14.3228</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>4536.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12.6215</td>\n",
       "      <td>14.2931</td>\n",
       "      <td>12.6612</td>\n",
       "      <td>13.0833</td>\n",
       "      <td>14.1782</td>\n",
       "      <td>13.8157</td>\n",
       "      <td>...</td>\n",
       "      <td>13.0353</td>\n",
       "      <td>14.6549</td>\n",
       "      <td>14.3492</td>\n",
       "      <td>14.2711</td>\n",
       "      <td>14.3810</td>\n",
       "      <td>14.2432</td>\n",
       "      <td>13.3940</td>\n",
       "      <td>12.8245</td>\n",
       "      <td>14.2491</td>\n",
       "      <td>14.3830</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198804</td>\n",
       "      <td>MT-CO1</td>\n",
       "      <td>4512.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12.6246</td>\n",
       "      <td>14.5995</td>\n",
       "      <td>12.5008</td>\n",
       "      <td>11.3334</td>\n",
       "      <td>15.3590</td>\n",
       "      <td>12.2692</td>\n",
       "      <td>...</td>\n",
       "      <td>13.3965</td>\n",
       "      <td>15.9901</td>\n",
       "      <td>15.2750</td>\n",
       "      <td>15.1726</td>\n",
       "      <td>14.8551</td>\n",
       "      <td>14.6084</td>\n",
       "      <td>13.5241</td>\n",
       "      <td>14.0931</td>\n",
       "      <td>15.5290</td>\n",
       "      <td>14.3970</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000198712</td>\n",
       "      <td>MT-CO2</td>\n",
       "      <td>4513.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>12.8384</td>\n",
       "      <td>14.7048</td>\n",
       "      <td>12.8456</td>\n",
       "      <td>12.6928</td>\n",
       "      <td>14.8495</td>\n",
       "      <td>13.4601</td>\n",
       "      <td>...</td>\n",
       "      <td>13.3606</td>\n",
       "      <td>15.3773</td>\n",
       "      <td>14.9873</td>\n",
       "      <td>14.6528</td>\n",
       "      <td>14.7760</td>\n",
       "      <td>14.8340</td>\n",
       "      <td>13.4387</td>\n",
       "      <td>14.1687</td>\n",
       "      <td>15.1614</td>\n",
       "      <td>14.3728</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000228253</td>\n",
       "      <td>MT-ATP8</td>\n",
       "      <td>4509.0</td>\n",
       "      <td>protein_coding</td>\n",
       "      <td>13.9763</td>\n",
       "      <td>15.7851</td>\n",
       "      <td>13.9252</td>\n",
       "      <td>14.1037</td>\n",
       "      <td>16.3241</td>\n",
       "      <td>15.3932</td>\n",
       "      <td>...</td>\n",
       "      <td>14.3118</td>\n",
       "      <td>16.4238</td>\n",
       "      <td>15.5251</td>\n",
       "      <td>15.8844</td>\n",
       "      <td>15.8104</td>\n",
       "      <td>15.4948</td>\n",
       "      <td>14.9172</td>\n",
       "      <td>14.8163</td>\n",
       "      <td>16.0166</td>\n",
       "      <td>15.2887</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows Ã— 7433 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ensembl_gene_id hgnc_symbol  entrezgene_id transcript_biotype  \\\n",
       "0  ENSG00000198888      MT-ND1         4535.0     protein_coding   \n",
       "1  ENSG00000198763      MT-ND2         4536.0     protein_coding   \n",
       "2  ENSG00000198804      MT-CO1         4512.0     protein_coding   \n",
       "3  ENSG00000198712      MT-CO2         4513.0     protein_coding   \n",
       "4  ENSG00000228253     MT-ATP8         4509.0     protein_coding   \n",
       "\n",
       "   GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
       "0                   13.1350                   14.6712   \n",
       "1                   12.6215                   14.2931   \n",
       "2                   12.6246                   14.5995   \n",
       "3                   12.8384                   14.7048   \n",
       "4                   13.9763                   15.7851   \n",
       "\n",
       "   GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
       "0                   12.8924                   12.6263   \n",
       "1                   12.6612                   13.0833   \n",
       "2                   12.5008                   11.3334   \n",
       "3                   12.8456                   12.6928   \n",
       "4                   13.9252                   14.1037   \n",
       "\n",
       "   GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  ...  \\\n",
       "0                   13.8339                   13.7962  ...   \n",
       "1                   14.1782                   13.8157  ...   \n",
       "2                   15.3590                   12.2692  ...   \n",
       "3                   14.8495                   13.4601  ...   \n",
       "4                   16.3241                   15.3932  ...   \n",
       "\n",
       "   GTEX-ZZPU-0726-SM-5N9C8  GTEX-ZZPU-0926-SM-5GZYT  GTEX-ZZPU-1226-SM-5N9CK  \\\n",
       "0                  12.8565                  14.8073                  14.3240   \n",
       "1                  13.0353                  14.6549                  14.3492   \n",
       "2                  13.3965                  15.9901                  15.2750   \n",
       "3                  13.3606                  15.3773                  14.9873   \n",
       "4                  14.3118                  16.4238                  15.5251   \n",
       "\n",
       "   GTEX-ZZPU-1326-SM-5GZWS  GTEX-ZZPU-1426-SM-5GZZ6  GTEX-ZZPU-1826-SM-5E43L  \\\n",
       "0                  14.0776                  14.1602                  13.7105   \n",
       "1                  14.2711                  14.3810                  14.2432   \n",
       "2                  15.1726                  14.8551                  14.6084   \n",
       "3                  14.6528                  14.7760                  14.8340   \n",
       "4                  15.8844                  15.8104                  15.4948   \n",
       "\n",
       "   GTEX-ZZPU-2126-SM-5EGIU  GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2626-SM-5E45Y  \\\n",
       "0                  12.8758                  12.5436                  14.6088   \n",
       "1                  13.3940                  12.8245                  14.2491   \n",
       "2                  13.5241                  14.0931                  15.5290   \n",
       "3                  13.4387                  14.1687                  15.1614   \n",
       "4                  14.9172                  14.8163                  16.0166   \n",
       "\n",
       "   GTEX-ZZPU-2726-SM-5NQ8O  \n",
       "0                  14.3228  \n",
       "1                  14.3830  \n",
       "2                  14.3970  \n",
       "3                  14.3728  \n",
       "4                  15.2887  \n",
       "\n",
       "[5 rows x 7433 columns]"
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "normal_protcoding.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {
    "id": "vXuPgJcIg916"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index(['sample', 'Sample_Type', 'Primary_Site', 'ATG7', 'ATG7_1', 'ATG7_2',\n",
      "       'ATG7_3', 'ATG7(total)', 'ATG7(1)', 'ATG7(2)', 'ATG7(3)',\n",
      "       'log2_+1_ATG7(total)', 'log2_+1_ATG7(1)', 'log2_+1_ATG7(2)',\n",
      "       'ATG7_prot_tot', 'ATG7_1%', 'ATG7_2%', 'ATG7_3%', 'ATG7(1)norm',\n",
      "       'ATG7(2)norm'],\n",
      "      dtype='object')\n",
      "-----------\n"
     ]
    },
    {
     "ename": "KeyError",
     "evalue": "'sample'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-32-eb39dab0bede>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m     29\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     30\u001b[0m   \u001b[0;31m#Merge the two df to have isoform expression + whole gene expression\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m---> 31\u001b[0;31m   \u001b[0mdf_merged\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mpandas\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mmerge\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;36m1\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mcurr_df\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mon\u001b[0m\u001b[0;34m=\u001b[0m \u001b[0;34m'sample'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     32\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     33\u001b[0m   \u001b[0mprint\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'df_isoform is: '\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mlen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mdf_Norm_Prim\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36mmerge\u001b[0;34m(left, right, how, on, left_on, right_on, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m    117\u001b[0m         \u001b[0mcopy\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mcopy\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    118\u001b[0m         \u001b[0mindicator\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mindicator\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 119\u001b[0;31m         \u001b[0mvalidate\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mvalidate\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    120\u001b[0m     )\n\u001b[1;32m    121\u001b[0m     \u001b[0;32mreturn\u001b[0m \u001b[0mop\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_result\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36m__init__\u001b[0;34m(self, left, right, how, on, left_on, right_on, axis, left_index, right_index, sort, suffixes, copy, indicator, validate)\u001b[0m\n\u001b[1;32m    697\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mright_join_keys\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    698\u001b[0m             \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin_names\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 699\u001b[0;31m         ) = self._get_merge_keys()\n\u001b[0m\u001b[1;32m    700\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    701\u001b[0m         \u001b[0;31m# validate the merge keys dtypes. We may need to coerce\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/reshape/merge.py\u001b[0m in \u001b[0;36m_get_merge_keys\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1107\u001b[0m                         \u001b[0mright_keys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1108\u001b[0m                     \u001b[0;32mif\u001b[0m \u001b[0mlk\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1109\u001b[0;31m                         \u001b[0mleft_keys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mleft\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_get_label_or_level_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1110\u001b[0m                         \u001b[0mjoin_names\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mappend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mlk\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1111\u001b[0m                     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/generic.py\u001b[0m in \u001b[0;36m_get_label_or_level_values\u001b[0;34m(self, key, axis)\u001b[0m\n\u001b[1;32m   1777\u001b[0m             \u001b[0mvalues\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maxes\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0maxis\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_level_values\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_values\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1778\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1779\u001b[0;31m             \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1780\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1781\u001b[0m         \u001b[0;31m# Check for duplicates\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'sample'"
     ]
    }
   ],
   "source": [
    "#Preparation of the global dataframe\n",
    "# data = [[df_protcoding[0], 'normal_protcoding'], [df_protcoding[1], 'primary_protcoding']]\n",
    "\n",
    "\n",
    "data = [[df_ATG7_Normal, normal_protcoding], [df_ATG7_Primary, primary_protcoding]]\n",
    "\n",
    "df_final = []\n",
    "\n",
    "for df in data:\n",
    "    \n",
    "  print(type(df[0]))\n",
    "  print(df[0].columns)\n",
    "  print('-----------')  \n",
    "\n",
    "  #print(df[0].columns()[:10])  \n",
    "  \n",
    "  #transpose the table \n",
    "  curr_df = df[0].copy()\n",
    "  curr_df = numpy.transpose(curr_df)\n",
    "\n",
    "  #Change head column & delete the name of the column index; choose ENSEMBL\n",
    "  curr_df.columns = curr_df.iloc[0]\n",
    "  curr_df.columns.name = None\n",
    "  curr_df = curr_df.reset_index()\n",
    "\n",
    "  #Delete the first rows & columns rename to merge\n",
    "  curr_df = curr_df.drop(curr_df.index[0:4])\n",
    "  curr_df.rename(columns = {'index' : 'sample'}, inplace = True)\n",
    "\n",
    "  #Merge the two df to have isoform expression + whole gene expression \n",
    "  df_merged = pandas.merge(df[1], curr_df, on= 'sample')\n",
    "\n",
    "  print('df_isoform is: ', len(df_Norm_Prim))\n",
    "  print('df_gene_exp is: ', len(df))\n",
    "  print('df_merged: ', len(df_merged))\n",
    "\n",
    "  #delete the sample column and select column\n",
    "  df_merged = df_merged.drop(\"sample\", 1)\n",
    "  df_merged = gexpr_compl.drop(gexpr_compl.columns[5:19], axis=1)\n",
    "  print(\"controler si j'ai bien ATG7_2\", 'suivi de ENSG00000198888')\n",
    "  \n",
    "  #save in list\n",
    "  df_final.append(df_merged)\n",
    "\n",
    "\n",
    "df_final[0] = df_normal_expr\n",
    "df_final[1] = df_primary_expr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "R8fLWHXOBYJl"
   },
   "source": [
    "# III] ATG7 expression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ufG3BGqPkmrk"
   },
   "source": [
    "##A) Distribution\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "vl1-Yk-8NVZi",
    "outputId": "227593fd-74c8-4670-8ad4-8a0ac27f1752"
   },
   "outputs": [],
   "source": [
    "# Distribution graph, for ATG7, iso1 & iso2\n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'in all tissue', df_ATG7_Primary, 'all primary tumors']]\n",
    "\n",
    "isoforms = [['log2_+1_ATG7(total)', 'ATG7'], ['log2_+1_ATG7(1)', 'ATG7(1)'], ['log2_+1_ATG7(2)', 'ATG7(2)']]\n",
    "\n",
    "\n",
    "for df in all_dfs:\n",
    "  for isoform in isoforms:\n",
    "    xData = df[0][isoform[0]]\n",
    "    x2Data = df[2][isoform[0]]\n",
    "\n",
    "    seaborn.kdeplot(xData, color = 'mediumaquamarine', fill = True)\n",
    "    seaborn.kdeplot(x2Data, color = 'sandybrown', fill = True)\n",
    "\n",
    "    plt.xlabel('log2 '+ isoform[1]+ ' +1')\n",
    "    plt.title(isoform[1]+ ' expression ' + df[1] )\n",
    "    plt.legend(labels=[\"Normal Tissue\",\"Primary Tumor \"])\n",
    "    plt.show()\n",
    "    plt.clf()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "PL_4aao-49o9"
   },
   "source": [
    "##B) Percentage do not express ATG7 isoforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "zwxExNKnmvhu",
    "outputId": "79575762-1bd2-4c3e-c35d-ede5886f24c6"
   },
   "outputs": [],
   "source": [
    "#Percentage of patients: \n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'all normal tissue'], [df_ATG7_Primary, 'all primary tumor']]\n",
    "\n",
    "isoforms = [['ATG7(total)', 'ATG7'], ['ATG7(1)', 'ATG7(1)'], ['ATG7(2)', 'ATG7(2)'], ['ATG7(3)', 'ATG7(3)']]\n",
    "\n",
    "for df in all_dfs:\n",
    "  print(df[1])\n",
    "  for isoform in isoforms:\n",
    "        \n",
    "    QUOI = isoform[1]\n",
    "    OU = df[1]\n",
    "    all = (df[0][isoform[0]]).shape\n",
    "    no_exp = (df[0][isoform[0]]<0.1).sum()\n",
    "    perc_no_exp = (no_exp*100)/all\n",
    "    perc_no_exp = float(perc_no_exp)\n",
    "\n",
    "    # print(\"The percentage of patient who do not express\", QUOI, OU, \"is:\", round(perc_no_exp,1),'%')\n",
    "    print(round(perc_no_exp,1),'% of tumor do not express', QUOI)\n",
    "  print()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "l9R88UMYsp9h"
   },
   "source": [
    "## C) Expression figures"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "j87nxfPMrMLB"
   },
   "source": [
    "### 1) Value TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 569
    },
    "id": "3_zvT1lweuYb",
    "outputId": "d561789f-e155-4e1d-ba5a-ac1a107c389a"
   },
   "outputs": [],
   "source": [
    "# Distribution: Data preparation + graph\n",
    "\n",
    "data = [[df_Norm_Prim, 'all tissue']]\n",
    "\n",
    "for df in data:\n",
    "   \n",
    "  shape = df[0].shape\n",
    "  shape = shape[0]\n",
    "  print(shape)\n",
    "\n",
    "  #select the sample type column\n",
    "  sample_type = {'sample_type':[df[0].iloc[:,1][i]\n",
    "      for i in range(0, shape)]}\n",
    "\n",
    "  #convert to df, and copy it 3 times\n",
    "  sample_typedf = pandas.DataFrame(sample_type)\n",
    "  sample_typedf = pandas.concat([sample_typedf,sample_typedf,sample_typedf])\n",
    "  sample_typedf.reset_index(drop=True, inplace=True)\n",
    "\n",
    "  # Y for expression value \n",
    "  y = {'y':[df[0].iloc[:,j][i]\n",
    "      for j in range(9,12)\n",
    "        for i in range(0, shape)]}\n",
    "\n",
    "  #create df with Y \n",
    "  data_distrib = pandas.DataFrame(y)\n",
    "\n",
    "  # X for ATG7/iso1/iso2\n",
    "  data_distrib['x'] = ''\n",
    "  data_distrib['x'][: shape] = 'ATG7'\n",
    "  data_distrib['x'][shape: shape*2] = 'ATG7(1)'\n",
    "  data_distrib['x'][shape*2:] = 'ATG7(2)'\n",
    "\n",
    "  # Will be used for Hue\n",
    "  data_distrib['Sample Type'] = sample_typedf['sample_type']\n",
    "\n",
    "  #graph\n",
    "  seaborn.catplot(x=\"x\", y=\"y\", hue=\"Sample Type\", aspect= 1.2, kind=\"box\", palette = 'Set2', data = data_distrib, \n",
    "                  boxprops={'lw':2}, medianprops={'lw':2}, whiskerprops={'lw':2}, showcaps=True, showfliers=False)\n",
    "  \n",
    "  plt.ylabel('Expression of ATG7 (TPM)')\n",
    "  plt.xlabel('')\n",
    "  plt.title('ATG7 expression in '+ df[1], x=.55)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "lqjTE5owk0cD",
    "outputId": "88756d11-ca02-4610-96f5-e52fc09ae047"
   },
   "outputs": [],
   "source": [
    "#Distribution: Stats for all \n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'all normal tissue', df_ATG7_Primary, 'all primary tumors']]\n",
    "\n",
    "isoforms = ['ATG7', 'ATG7_1', 'ATG7_2']\n",
    "\n",
    "for df in all_dfs:\n",
    "  for isoform in isoforms:\n",
    "\n",
    "    n1 = \"{:.3f}\".format(numpy.median(df[0][isoform]))\n",
    "    n2 = \"{:.3f}\".format(numpy.median(df[2][isoform]))\n",
    "\n",
    "    # Mann-Whitney analysis \n",
    "    statistic, pvalue = scipy.stats.mannwhitneyu(df[0][isoform], df[2][isoform])\n",
    "    Pvalue = \"{:.3E}\".format(pvalue)\n",
    "    print(\"the pvalue between\", df[1]+'('+n1+')', \"and\", df[3]+'('+n2+')', \"for\", isoform, 'is', Pvalue)\n",
    "  print()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "rccgpnCWrUyE"
   },
   "source": [
    "### 2) Normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 722
    },
    "id": "n_h00in7pPDk",
    "outputId": "90718d06-95e5-4f40-8c1e-e18cb5a259a3"
   },
   "outputs": [],
   "source": [
    "# NORMALiZED Distribution: Data preparation + graph\n",
    "\n",
    "data = [[df_Norm_Prim, 'all_tissue']]\n",
    "\n",
    "\n",
    "for df in data:\n",
    "   \n",
    "  shape = df[0].shape\n",
    "  shape = shape[0]\n",
    "  print(shape)\n",
    "\n",
    "  #select the sample type column\n",
    "  sample_type = {'sample_type':[df[0].iloc[:,1][i]\n",
    "      for i in range(0, shape)]}\n",
    "\n",
    "  #convert to df, and copy it 3 times\n",
    "  sample_typedf = pandas.DataFrame(sample_type)\n",
    "  sample_typedf = pandas.concat([sample_typedf,sample_typedf,sample_typedf])\n",
    "  sample_typedf.reset_index(drop=True, inplace=True)\n",
    "\n",
    "  # Y for expression value \n",
    "  y = {'y':[df[0].iloc[:,j][i]\n",
    "      for j in range(18,20)\n",
    "        for i in range(0, shape)]}\n",
    "\n",
    "  #create df with Y \n",
    "  data_distrib = pandas.DataFrame(y)\n",
    "\n",
    "  # X for iso1/ATG7 and iso2/ATG7\n",
    "  data_distrib['x'] = ''\n",
    "  data_distrib['x'][: shape] = 'ATG7(1)/ATG7'\n",
    "  data_distrib['x'][shape:] = 'ATG7(2)/ATG7'\n",
    "\n",
    "\n",
    "  # Will be used for Hue\n",
    "  data_distrib['Sample Type'] = sample_typedf['sample_type']\n",
    "\n",
    "  #graph\n",
    "  seaborn.catplot(x=\"x\", y=\"y\", hue=\"Sample Type\", aspect= 1.2, kind=\"box\", palette = 'Set2', data = data_distrib, \n",
    "                  boxprops={'lw':2}, medianprops={'lw':2}, whiskerprops={'lw':2}, showcaps=True, showfliers=False)\n",
    "  \n",
    "  plt.ylabel('ATG7 isoforms/total ATG7 in TPM')\n",
    "  plt.xlabel('')\n",
    "  plt.title('ATG7 proportion in '+ df[1], x=0.55)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "JflZDsJbpPDl",
    "outputId": "064d15f1-d035-4b7b-f4d5-bf8e473a8518"
   },
   "outputs": [],
   "source": [
    "#NORMALIZED Distribution: Stats for all \n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'all normal tissue', df_ATG7_Primary, 'all primary tumors']]\n",
    "\n",
    "isoforms = ['ATG7(1)norm', 'ATG7(2)norm']\n",
    "print('NORMALIZED')\n",
    "\n",
    "for df in all_dfs:\n",
    "  for isoform in isoforms:\n",
    "\n",
    "    n1 = \"{:.3f}\".format(numpy.median(df[0][isoform]))\n",
    "    n2 = \"{:.3f}\".format(numpy.median(df[2][isoform]))\n",
    "\n",
    "    # Mann-Whitney analysis \n",
    "    statistic, pvalue = scipy.stats.mannwhitneyu(df[0][isoform], df[2][isoform])\n",
    "    Pvalue = \"{:.3E}\".format(pvalue)\n",
    "    print(\"the pvalue between\", df[1]+'('+n1+')', \"and\", df[3]+'('+n2+')', \"for\", isoform, 'is', Pvalue)\n",
    "  print()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ivLjI2vHTV5A"
   },
   "source": [
    "#IV] Expression correlation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "4azGSbHX1l11",
    "outputId": "8e6361eb-8c82-479d-c71e-2bf6fb6c743e"
   },
   "outputs": [],
   "source": [
    "#isoform correlation in all tissue \n",
    "\n",
    "#three values are very high and make my graph ugly... I remove them.\n",
    "df_ATG7_Primary2 = Primary[Primary['ATG7(1)'] < 30 ]\n",
    "print(Primary.shape)\n",
    "print(Primary2.shape)\n",
    "\n",
    "\n",
    "tissues = [[df_ATG7_Normal, \"normal tissue\"], [df_ATG7_Primary2, \"primary tumors\"]]\n",
    "\n",
    "for tissue in tissues: \n",
    "  x = tissue[0]['ATG7(1)'].to_list()\n",
    "  y = tissue[0]['ATG7(2)'].to_list()\n",
    "\n",
    "  r_value, p_value = scipy.stats.spearmanr(x, y)\n",
    "  print(r_value, p_value)\n",
    "\n",
    "  seaborn.regplot(x=x,y=y, x_bins=800, x_ci = 0, marker=\"+\")\n",
    "  name = \"Correlation between the two isoforms in \"+  tissue[1]\n",
    "  plt.title(name)\n",
    "  plt.xlabel(\"ATG7(1) expression in tpm\")\n",
    "  plt.ylabel('ATG7(2) expression tpm')\n",
    "  plt.show()\n",
    "  plt.clf()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "1Pspjsp_10nE"
   },
   "outputs": [],
   "source": [
    "#To see information. \n",
    "# â†’ I can see many samples have no expression of ATG7(2) so I loose many data in normal tissue when removing <4tpm.\n",
    "#If I want to work on Normal, I have to do exeption on iso2\n",
    "\n",
    "all_tissue_used = [[df_normal_exp,'Normal Tissue'], [df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "isoforms = ['ATG7_1', 'ATG7_2']\n",
    "\n",
    "for tissue in all_tissue_used:\n",
    "  print(tissue[1])\n",
    "  ff = tissue[0].shape\n",
    "  print('number of patients:', ff[0])\n",
    "  print(\"ATG7_1 max is\", tissue[0]['ATG7_1'].max())\n",
    "  print(\"ATG7_2 max is\", tissue[0]['ATG7_2'].max())\n",
    "    \n",
    "\n",
    "  for isoform in isoforms:\n",
    "    print('shape before removing', isoform, 'zero expression', tissue[0].shape)\n",
    "    df_atleast1 = tissue[0][tissue[0][isoform]>=0]\n",
    "    print('shape after removing', isoform, 'zero expression', df_atleast1.shape)\n",
    "\n",
    "    #create a new row with max() for each gene\n",
    "    tissue[0].loc['max()'] = tissue[0].max()\n",
    "\n",
    "    #Transpose to have max() in column\n",
    "    tissue[0] = tissue[0].T\n",
    "    nb_before = tissue[0].shape\n",
    "    tissue[0] = tissue[0][tissue[0]['max()'] > 1.2]\n",
    "    nb_after = tissue[0].shape\n",
    "    print(\"number of genes removed:\", nb_before[0]-nb_after[0])\n",
    "    print('now the minimum of TPM is', tissue[0]['max()'].min())\n",
    "\n",
    "    #delete the column max()\n",
    "    tissue[0] =  tissue[0].drop('max()', axis = 1) \n",
    "    # print(tissue[0].shape)\n",
    "\n",
    "    #transpose back\n",
    "    tissue[0] = tissue[0].T\n",
    "\n",
    "\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "LxKzIf-63QkM"
   },
   "outputs": [],
   "source": [
    "#to know the ensembl code for HK2\n",
    "protein_coding_list.loc[protein_coding_list['hgnc_symbol'] == 'MFAP3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "hsgaqjKl3z5F"
   },
   "outputs": [],
   "source": [
    "# Spearman Correlation, main genes \n",
    "\n",
    "all_tissue_used = [[df_normal_exp,'Normal Tissue'], [df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "HK2 = 'ENSG00000159399'\n",
    "YAP1 = 'ENSG00000137693'\n",
    "AJUBA = 'ENSG00000129474'\n",
    "YBX1 = 'ENSG00000065978'\n",
    "TGFB1 = 'ENSG00000105329'\n",
    "ATG7_1 = 'ATG7_1'\n",
    "ATG7_2 = 'ATG7_2'\n",
    "MTDH = 'ENSG00000147649'\n",
    "MAPK1 = 'ENSG00000100030'\n",
    "KPNA1 = 'ENSG00000114030'\n",
    "TNPO1 = 'ENSG00000083312'\n",
    "\n",
    "for tissue in all_tissue_used:\n",
    "  genes_x = [[KPNA1,'KPNA1'], [MAPK1, 'MAPK1'], [TNPO1,'TNPO1'], [MTDH,'MTDH'], \n",
    "             [HK2, \"HK2\"], [YAP1, 'YAP1'], [AJUBA,'AJUBA'], [YBX1,'YBX1'], \n",
    "             [TGFB1,'TGFB1'], [ATG7_1,'ATG7_1'], [ATG7_2,'ATG7_2']]\n",
    "\n",
    "  print(tissue[1])\n",
    "\n",
    "  for gene in genes_x:\n",
    "    x = tissue[0][gene[0]].to_list()\n",
    "    y = tissue[0]['ATG7'].to_list()\n",
    "    y1 = tissue[0]['ATG7_1'].to_list()\n",
    "    y2 = tissue[0]['ATG7_2'].to_list()  \n",
    "  \n",
    "    r_value, p_value = scipy.stats.spearmanr(x, y)\n",
    "    print('  correlation between ATG7 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "    r_value, p_value = scipy.stats.spearmanr(x, y1)\n",
    "    print('  correlation between ATG7_1 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "    r_value, p_value = scipy.stats.spearmanr(x, y2)\n",
    "    print('  correlation between ATG7_2 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "DSfvcpoN4r9O"
   },
   "outputs": [],
   "source": [
    "# Spearman Correlation, all genes, PRIMARY TUMORS\n",
    "all_tissue_used = [[df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "isoforms = ['ATG7_1', 'ATG7_2']\n",
    "\n",
    "all_positiv_corr_spearman = []\n",
    "\n",
    "for tissue in all_tissue_used:\n",
    "  print(tissue[1], tissue[0].shape)\n",
    "\n",
    "  #create a new row with max() for each gene and transpose\n",
    "  tissue[0].loc['max()'] = tissue[0].max()\n",
    "  tissue[0] = tissue[0].T\n",
    "\n",
    "  #select all row where max() > 2.3 (= 5 tpm)\n",
    "  nb_before = tissue[0]['max()'].count()\n",
    "  tissue[0] = tissue[0][tissue[0]['max()'] > 2.3]\n",
    "  nb_after = tissue[0]['max()'].count()\n",
    "  print(\"number of genes removed:\", nb_before-nb_after)\n",
    "  print('the minimum of TPM is', tissue[0]['max()'].min())\n",
    "  \n",
    "  #delete the column max() and tranpose back\n",
    "  tissue[0] =  tissue[0].drop('max()', axis = 1) \n",
    "  tissue[0] = tissue[0].T\n",
    "  print('nombre genes after removing < 4tpm', tissue[0].shape)\n",
    "\n",
    "  for isoform in isoforms:\n",
    "\n",
    "    genepos = []\n",
    "    geneneg = []\n",
    "    rvaluepos = []\n",
    "    rvalueneg = []\n",
    "    pvaluepos = []\n",
    "    pvalueneg = []\n",
    "\n",
    "    print(isoform)\n",
    "    x = tissue[0][isoform].to_list()\n",
    "    genes = tissue[0].columns[3:]\n",
    "\n",
    "    for gene in genes:\n",
    "      y = tissue[0][gene].to_list()\n",
    "      r_value, p_value = scipy.stats.spearmanr(x, y)\n",
    "\n",
    "      if r_value > 0.65:\n",
    "        genepos.append(gene)\n",
    "        rvaluepos.append(r_value)\n",
    "        pvaluepos.append(p_value)\n",
    "        \n",
    "      elif r_value < -0.65:\n",
    "        geneneg.append(gene)\n",
    "        rvalueneg.append(r_value)\n",
    "        pvalueneg.append(p_value)\n",
    "\n",
    "    #create a dictionnary\n",
    "    d = dict(ensembl_gene_id= genepos, r_value=rvaluepos,  p_value=pvaluepos)\n",
    "    d2 = dict(Gene= geneneg, r_value=rvalueneg,  p_value=pvalueneg)\n",
    "\n",
    "    #Convert to dataframe\n",
    "    positiv_corr = 'pos_corr_'+tissue[1]\n",
    "    negativ_corr = 'neg_corr_'+tissue[1]\n",
    "\n",
    "    df_positiv_corr = pandas.DataFrame.from_dict(d, orient='index')\n",
    "    df_negativ_corr = pandas.DataFrame.from_dict(data=d2, orient='index')\n",
    "\n",
    "    #Transpose to have a better table\n",
    "    df_positiv_corr = df_positiv_corr.transpose()\n",
    "    df_negativ_corr = df_negativ_corr.transpose()\n",
    "\n",
    "    #name\n",
    "    positiv_corr = 'pos_spearman_corr_'+tissue[1]+'_'+isoform\n",
    "    negativ_corr = 'neg_spearman_corr_'+tissue[1]+'_'+isoform\n",
    "\n",
    "    print(positiv_corr, df_positiv_corr.shape)\n",
    "    print(negativ_corr, df_negativ_corr.shape)\n",
    "\n",
    "    #scending orde, save to excel\n",
    "    df_positiv_corr['info'] = tissue[1]+'_'+isoform\n",
    "    df_positiv_corr = protein_coding_list.merge(df_positiv_corr)\n",
    "    df_positiv_corr = df_positiv_corr.drop(['entrezgene_id', 'transcript_biotype'], axis = 1)\n",
    "\n",
    "    df_positiv_corr = df_positiv_corr.sort_values(['r_value'], ascending = False)\n",
    "\n",
    "    df_positiv_corr.to_excel(positiv_corr+'.xlsx', index=False)\n",
    "    df_negativ_corr.to_excel(negativ_corr+'.xlsx', index=False)\n",
    "\n",
    "    all_positiv_corr_spearman.append(df_positiv_corr)\n",
    "  print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "OGqyJk-14Zyq"
   },
   "outputs": [],
   "source": [
    "#Graphic representation, two graphs in one\n",
    "\n",
    "all_tissue_used = [[df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "HK2 = 'ENSG00000159399'\n",
    "YAP1 = 'ENSG00000137693'\n",
    "AJUBA = 'ENSG00000129474'\n",
    "YBX1 = 'ENSG00000065978'\n",
    "TGFB1 = 'ENSG00000105329'\n",
    "MAPK1 = 'ENSG00000100030'\n",
    "MTDH = 'ENSG00000147649'\n",
    "KPNA1 = 'ENSG00000114030'\n",
    "TNPO1 = 'ENSG00000083312'\n",
    "\n",
    "\n",
    " \n",
    "genes = [[ATG7,'ATG7'], [ATG7_1,'ATG7_1'], [MAPK1,'MAPK1'], [MTDH,'MTDH'], [KPNA1,'KPNA1'], \n",
    "         [TNPO1,'TNPO1'], [HK2,'HK2'], [YAP1,'YAP1'], [AJUBA,'AJUBA'], [YBX1,'YBX1'], [TGFB1,'TGFB1']]\n",
    "\n",
    "for gene in genes:\n",
    "  for df in data:\n",
    "    \n",
    "      fig, ax = plt.subplots(1,2, figsize=(18,8))\n",
    "      name = gene[1]+ \" correlation with the two isoforms\"\n",
    "      name2 = gene[1]+ \" expression (log2 tpm)\"\n",
    "\n",
    "      fig.suptitle(name, fontsize=\"x-large\")\n",
    "\n",
    "      x = df[0][gene[0]].to_list()\n",
    "      y1 = df[0]['ATG7_1'].to_list()\n",
    "      y2 = df[0]['ATG7_2'].to_list()\n",
    "\n",
    "\n",
    "      fig = seaborn.regplot(x=x,y=y1, ax=ax[0], x_bins=500, label = df[1], color = df[2])\n",
    "      fig.set_title(\"ATG7(1)\")\n",
    "      fig.set_xlabel(name2)\n",
    "      fig.set_ylabel('ATG7(1) expression (log2 tpm)')\n",
    "\n",
    "      fig = seaborn.regplot(x=x,y=y2, ax=ax[1], x_bins=500, label = df[1], color = df[2])\n",
    "      fig.set_title(\"ATG7(2)\")\n",
    "      fig.set_xlabel(name2)\n",
    "      fig.set_ylabel('ATG7(2) expression (log2 tpm)')\n",
    "      plt.legend()\n",
    "      plt.show()\n",
    "      plt.clf()\n",
    "\n",
    "      print(df[1])\n",
    "      r_value, p_value = scipy.stats.spearmanr(x, y1)\n",
    "      print('  correlation between ATG7_1 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "      r_value, p_value = scipy.stats.spearmanr(x, y2)\n",
    "      print('  correlation between ATG7_2 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "      print()"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [
    "CGVz9_mqWgpw"
   ],
   "name": "all_organs.ipynb",
   "provenance": []
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
