{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {
    "id": "TLQwwG5VDPFA"
   },
   "outputs": [],
   "source": [
    "import pandas, numpy, math, seaborn\n",
    "import scipy, scipy.stats"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {
    "id": "CMOWiLBfDPFA"
   },
   "outputs": [],
   "source": [
    "# module for figures\n",
    "import matplotlib, matplotlib.pyplot as plt\n",
    "\n",
    "# this is a trick to make figures look nicer\n",
    "matplotlib.rcParams.update({'font.size':20, 'font.family':'FreeSans', 'xtick.labelsize':20, 'ytick.labelsize':10, 'figure.figsize':(12, 8)})"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {
    "id": "jmnIMX-FfoyD"
   },
   "outputs": [],
   "source": [
    "pandas.set_option(\"max_rows\", None)\n",
    "# pandas.reset_option(\"max_rows\")"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# 0. user-defined variables"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {
    "id": "aHG0vowTDPFC"
   },
   "outputs": [],
   "source": [
    "input_file_directory = '/home/adrian/projects/ATG7/data/'\n",
    "\n",
    "normal_tissue_expression_file = '/home/adrian/projects/ATG7/results/normal_tissue_expression.tsv'\n",
    "primary_tumor_expression_file = '/home/adrian/projects/ATG7/results/primary_tumors_expression.tsv'"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "wYGSc-xPDaFV"
   },
   "source": [
    "## functions & options\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {
    "id": "K8VbSwcZyDon"
   },
   "outputs": [],
   "source": [
    "#define a function to split after the \".\". The 1 is for how many word to have. if we have several dot. The [1] is what is after the dot, and [0] what is before\n",
    "def split(L):\n",
    "  return L.split(\".\",1)[0]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "id": "ccNktmiu2K2n"
   },
   "outputs": [],
   "source": [
    "def binarise_mutation(DF):\n",
    "  old = ['no variant', \"3'UTR\", \"5'UTR\", 'Intron', \"5'Flank\", 'Silent', \n",
    "         'large deletion', 'Nonstop_Mutation','Missense_Mutation', 'Nonsense_Mutation', 'Frame_Shift_Del', 'Frame_Shift_Ins', 'Splice_Site', 'In_Frame_Del', 'In_Frame_Ins', 'Translation_Start_Site']\n",
    "  new = [0,0,0,0,0,0,\n",
    "         1,1,1,1,1,1,1,1,1,1]\n",
    "\n",
    "  for i in range(len(old)):\n",
    "    DF['effect'] = DF['effect'].replace(old[i], new[i])\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "IpmCT5mEDPFD"
   },
   "source": [
    "# II] Dataframe settings"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "scwtwuDCqUKz"
   },
   "source": [
    "## A) Isoform data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "id": "PHqx_s0JozXU"
   },
   "outputs": [],
   "source": [
    "# Original data isoforms\n",
    "directory = input_file_directory + \"xena_surv_ATG7.tsv\"\n",
    "df_ori= pandas.read_csv(directory, sep = \"\\t\")\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "id": "oC8hQXN9WRsH",
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  # This is added back by InteractiveShellApp.init_path()\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py:5047: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  errors=errors,\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:23: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:24: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:25: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:26: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:29: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:30: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:31: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:34: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:35: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:36: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:37: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:40: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:41: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ATG7_3</th>\n",
       "      <th>ATG7(total)</th>\n",
       "      <th>ATG7(1)</th>\n",
       "      <th>ATG7(2)</th>\n",
       "      <th>ATG7(3)</th>\n",
       "      <th>log2_+1_ATG7(total)</th>\n",
       "      <th>log2_+1_ATG7(1)</th>\n",
       "      <th>log2_+1_ATG7(2)</th>\n",
       "      <th>ATG7_prot_tot</th>\n",
       "      <th>ATG7_1%</th>\n",
       "      <th>ATG7_2%</th>\n",
       "      <th>ATG7_3%</th>\n",
       "      <th>ATG7(1)norm</th>\n",
       "      <th>ATG7(2)norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>TARGET-20-PARUBT-40</td>\n",
       "      <td>Recurrent Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>4.401</td>\n",
       "      <td>1.3900</td>\n",
       "      <td>1.522</td>\n",
       "      <td>-1.7320</td>\n",
       "      <td>21.125765</td>\n",
       "      <td>2.619787</td>\n",
       "      <td>2.870889</td>\n",
       "      <td>3.000343e-01</td>\n",
       "      <td>4.467655</td>\n",
       "      <td>1.855905</td>\n",
       "      <td>1.952665</td>\n",
       "      <td>5.790710</td>\n",
       "      <td>45.241201</td>\n",
       "      <td>49.577495</td>\n",
       "      <td>5.181305</td>\n",
       "      <td>0.124009</td>\n",
       "      <td>0.135895</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>TARGET-20-PATJHJ-40</td>\n",
       "      <td>Recurrent Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>2.820</td>\n",
       "      <td>1.1180</td>\n",
       "      <td>-1.086</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>7.060624</td>\n",
       "      <td>2.169459</td>\n",
       "      <td>0.470066</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>3.010892</td>\n",
       "      <td>1.664236</td>\n",
       "      <td>0.555881</td>\n",
       "      <td>2.639524</td>\n",
       "      <td>82.191280</td>\n",
       "      <td>17.808726</td>\n",
       "      <td>-0.000006</td>\n",
       "      <td>0.307262</td>\n",
       "      <td>0.066576</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>TARGET-10-PASLZM-40</td>\n",
       "      <td>Recurrent Blood Derived Cancer - Peripheral Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>1.158</td>\n",
       "      <td>0.0158</td>\n",
       "      <td>-3.171</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>2.230479</td>\n",
       "      <td>1.010012</td>\n",
       "      <td>0.110028</td>\n",
       "      <td>7.000183e-02</td>\n",
       "      <td>1.691748</td>\n",
       "      <td>1.007204</td>\n",
       "      <td>0.150597</td>\n",
       "      <td>1.190042</td>\n",
       "      <td>84.871949</td>\n",
       "      <td>9.245753</td>\n",
       "      <td>5.882299</td>\n",
       "      <td>0.452823</td>\n",
       "      <td>0.049329</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>TARGET-21-PATAIJ-42</td>\n",
       "      <td>Post treatment Blood Cancer - Blood</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>4.332</td>\n",
       "      <td>1.3450</td>\n",
       "      <td>1.975</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>20.139115</td>\n",
       "      <td>2.539302</td>\n",
       "      <td>3.930282</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>4.401843</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>2.301670</td>\n",
       "      <td>6.469584</td>\n",
       "      <td>39.249848</td>\n",
       "      <td>60.750154</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>0.126088</td>\n",
       "      <td>0.195157</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>TARGET-21-PASVJS-41</td>\n",
       "      <td>Post treatment Blood Cancer - Bone Marrow</td>\n",
       "      <td>White Blood Cell</td>\n",
       "      <td>5.156</td>\n",
       "      <td>1.8040</td>\n",
       "      <td>3.769</td>\n",
       "      <td>0.5069</td>\n",
       "      <td>35.653197</td>\n",
       "      <td>3.490870</td>\n",
       "      <td>13.631706</td>\n",
       "      <td>1.419994e+00</td>\n",
       "      <td>5.195867</td>\n",
       "      <td>2.166995</td>\n",
       "      <td>3.871026</td>\n",
       "      <td>18.542569</td>\n",
       "      <td>18.826249</td>\n",
       "      <td>73.515731</td>\n",
       "      <td>7.658019</td>\n",
       "      <td>0.097912</td>\n",
       "      <td>0.382342</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                sample                                        Sample_Type  \\\n",
       "0  TARGET-20-PARUBT-40  Recurrent Blood Derived Cancer - Peripheral Blood   \n",
       "1  TARGET-20-PATJHJ-40  Recurrent Blood Derived Cancer - Peripheral Blood   \n",
       "2  TARGET-10-PASLZM-40  Recurrent Blood Derived Cancer - Peripheral Blood   \n",
       "3  TARGET-21-PATAIJ-42                Post treatment Blood Cancer - Blood   \n",
       "4  TARGET-21-PASVJS-41          Post treatment Blood Cancer - Bone Marrow   \n",
       "\n",
       "       Primary_Site   ATG7  ATG7_1  ATG7_2  ATG7_3  ATG7(total)   ATG7(1)  \\\n",
       "0  White Blood Cell  4.401  1.3900   1.522 -1.7320    21.125765  2.619787   \n",
       "1  White Blood Cell  2.820  1.1180  -1.086 -9.9660     7.060624  2.169459   \n",
       "2  White Blood Cell  1.158  0.0158  -3.171 -3.8160     2.230479  1.010012   \n",
       "3  White Blood Cell  4.332  1.3450   1.975 -9.9660    20.139115  2.539302   \n",
       "4  White Blood Cell  5.156  1.8040   3.769  0.5069    35.653197  3.490870   \n",
       "\n",
       "     ATG7(2)       ATG7(3)  log2_+1_ATG7(total)  log2_+1_ATG7(1)  \\\n",
       "0   2.870889  3.000343e-01             4.467655         1.855905   \n",
       "1   0.470066 -1.495113e-07             3.010892         1.664236   \n",
       "2   0.110028  7.000183e-02             1.691748         1.007204   \n",
       "3   3.930282 -1.495113e-07             4.401843         1.823465   \n",
       "4  13.631706  1.419994e+00             5.195867         2.166995   \n",
       "\n",
       "   log2_+1_ATG7(2)  ATG7_prot_tot    ATG7_1%    ATG7_2%   ATG7_3%  \\\n",
       "0         1.952665       5.790710  45.241201  49.577495  5.181305   \n",
       "1         0.555881       2.639524  82.191280  17.808726 -0.000006   \n",
       "2         0.150597       1.190042  84.871949   9.245753  5.882299   \n",
       "3         2.301670       6.469584  39.249848  60.750154 -0.000002   \n",
       "4         3.871026      18.542569  18.826249  73.515731  7.658019   \n",
       "\n",
       "   ATG7(1)norm  ATG7(2)norm  \n",
       "0     0.124009     0.135895  \n",
       "1     0.307262     0.066576  \n",
       "2     0.452823     0.049329  \n",
       "3     0.126088     0.195157  \n",
       "4     0.097912     0.382342  "
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Column selection\n",
    "df= df_ori[['sample',\n",
    "            '_sample_type',\n",
    "            '_primary_site',\n",
    "            'ENSG00000197548.12',\n",
    "            'ENST00000354449.7', \n",
    "            'ENST00000354956.9',\n",
    "            'ENST00000446450.6']]\n",
    "\n",
    "# We have Adrenal Gland and Adrenal gland. I need to put \"G\".\n",
    "df['_primary_site'] = df['_primary_site'].str.title()\n",
    "\n",
    "# Column rename\n",
    "df.rename(columns = {'sample' : 'sample',\n",
    "    '_sample_type' : 'Sample_Type', \n",
    "                            '_primary_site' : 'Primary_Site', \n",
    "                            'ENST00000354449.7': 'ATG7_1',\n",
    "                            'ENST00000354956.9': 'ATG7_2',\n",
    "                            'ENST00000446450.6': 'ATG7_3',\n",
    "                            'ENSG00000197548.12': 'ATG7'}, inplace = True)\n",
    "\n",
    "# Calcul to have value from log2;  2**(ATG7(1))-0.001\n",
    "df[\"ATG7(total)\"] = pow(2,df['ATG7'])-0.001\n",
    "df[\"ATG7(1)\"] = pow(2,df['ATG7_1'])-0.001\n",
    "df[\"ATG7(2)\"] = pow(2,df['ATG7_2'])-0.001\n",
    "df[\"ATG7(3)\"] = pow(2,df['ATG7_3'])-0.001\n",
    "\n",
    "# Calcul to have log2+1 from value; \n",
    "df[\"log2_+1_ATG7(total)\"] = numpy.log2(df['ATG7(total)']+1)\n",
    "df[\"log2_+1_ATG7(1)\"] = numpy.log2(df['ATG7(1)']+1)\n",
    "df[\"log2_+1_ATG7(2)\"] = numpy.log2(df['ATG7(2)']+1)\n",
    "\n",
    "# Calcul to see percentage of expression of the 3 isoforms of ATG7\n",
    "df['ATG7_prot_tot'] = (df['ATG7(1)'] + df['ATG7(2)'] + df['ATG7(3)'])\n",
    "df['ATG7_1%'] = (df['ATG7(1)'] *100)/ df['ATG7_prot_tot']\n",
    "df['ATG7_2%'] = (df['ATG7(2)'] *100)/ df['ATG7_prot_tot']\n",
    "df['ATG7_3%'] = (df['ATG7(3)'] *100)/ df['ATG7_prot_tot']\n",
    "\n",
    "#Add columns, normalized on total\n",
    "df[\"ATG7(1)norm\"] = df['ATG7(1)']/df[\"ATG7(total)\"]\n",
    "df[\"ATG7(2)norm\"] = df['ATG7(2)']/df[\"ATG7(total)\"]\n",
    "df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "l2tWUsTWX8ES",
    "outputId": "1ed530fb-9179-4afd-a95f-2d36353eef32"
   },
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Primary Tumor                                        9185\n",
       "Normal Tissue                                        7429\n",
       "Solid Tissue Normal                                   738\n",
       "Cell Line                                             433\n",
       "Metastatic                                            393\n",
       "Primary Solid Tumor                                   286\n",
       "Primary Blood Derived Cancer - Peripheral Blood       239\n",
       "Primary Blood Derived Cancer - Bone Marrow            237\n",
       "Recurrent Blood Derived Cancer - Bone Marrow          104\n",
       "Recurrent Tumor                                        45\n",
       "Recurrent Solid Tumor                                  13\n",
       "Post treatment Blood Cancer - Bone Marrow              12\n",
       "Additional - New Primary                               11\n",
       "Recurrent Blood Derived Cancer - Peripheral Blood       3\n",
       "Post treatment Blood Cancer - Blood                     1\n",
       "Control Analyte                                         1\n",
       "Additional Metastatic                                   1\n",
       "Name: Sample_Type, dtype: int64"
      ]
     },
     "execution_count": 9,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df['Sample_Type'].value_counts()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "kteWDMcBRLe1",
    "outputId": "609ba62e-868c-4942-dd33-728803f4ca7c"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "normal_tissue 7429\n",
      "primary_tumors 9185\n"
     ]
    }
   ],
   "source": [
    "#Creation of two tables, Normal tissue and Primary tumor.\n",
    "df_ATG7_Normal = df[df['Sample_Type'].str.match ('Normal Tissue')]\n",
    "df_ATG7_Primary = df[df['Sample_Type'].str.match ('Primary Tumor')]\n",
    "\n",
    "print('normal_tissue' , len(df_ATG7_Normal))\n",
    "print('primary_tumors' , len(df_ATG7_Primary))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 366
    },
    "id": "90ZnSHOcSa--",
    "outputId": "a04c738c-d866-42e0-ff86-5ae0fce970ac"
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>sample</th>\n",
       "      <th>Sample_Type</th>\n",
       "      <th>Primary_Site</th>\n",
       "      <th>ATG7</th>\n",
       "      <th>ATG7_1</th>\n",
       "      <th>ATG7_2</th>\n",
       "      <th>ATG7_3</th>\n",
       "      <th>ATG7(total)</th>\n",
       "      <th>ATG7(1)</th>\n",
       "      <th>ATG7(2)</th>\n",
       "      <th>ATG7(3)</th>\n",
       "      <th>log2_+1_ATG7(total)</th>\n",
       "      <th>log2_+1_ATG7(1)</th>\n",
       "      <th>log2_+1_ATG7(2)</th>\n",
       "      <th>ATG7_prot_tot</th>\n",
       "      <th>ATG7_1%</th>\n",
       "      <th>ATG7_2%</th>\n",
       "      <th>ATG7_3%</th>\n",
       "      <th>ATG7(1)norm</th>\n",
       "      <th>ATG7(2)norm</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GTEX-ZTTD-0326-SM-57WFW</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>5.042</td>\n",
       "      <td>3.860</td>\n",
       "      <td>2.390</td>\n",
       "      <td>0.2642</td>\n",
       "      <td>32.944283</td>\n",
       "      <td>14.519306</td>\n",
       "      <td>5.240574</td>\n",
       "      <td>1.199970e+00</td>\n",
       "      <td>5.085097</td>\n",
       "      <td>3.955992</td>\n",
       "      <td>2.641679</td>\n",
       "      <td>20.959850</td>\n",
       "      <td>69.271996</td>\n",
       "      <td>25.002916</td>\n",
       "      <td>5.725088</td>\n",
       "      <td>0.440723</td>\n",
       "      <td>0.159074</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GTEX-PX3G-1626-SM-2S1PT</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>4.252</td>\n",
       "      <td>2.322</td>\n",
       "      <td>1.956</td>\n",
       "      <td>1.1320</td>\n",
       "      <td>19.052710</td>\n",
       "      <td>4.999249</td>\n",
       "      <td>3.878848</td>\n",
       "      <td>2.190624e+00</td>\n",
       "      <td>4.325725</td>\n",
       "      <td>2.584782</td>\n",
       "      <td>2.286540</td>\n",
       "      <td>11.068720</td>\n",
       "      <td>45.165557</td>\n",
       "      <td>35.043325</td>\n",
       "      <td>19.791118</td>\n",
       "      <td>0.262390</td>\n",
       "      <td>0.203585</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GTEX-OXRO-1726-SM-3LK6C</td>\n",
       "      <td>Normal Tissue</td>\n",
       "      <td>Muscle</td>\n",
       "      <td>3.909</td>\n",
       "      <td>2.602</td>\n",
       "      <td>1.345</td>\n",
       "      <td>-9.9660</td>\n",
       "      <td>15.020948</td>\n",
       "      <td>6.070277</td>\n",
       "      <td>2.539302</td>\n",
       "      <td>-1.495113e-07</td>\n",
       "      <td>4.001888</td>\n",
       "      <td>2.821767</td>\n",
       "      <td>1.823465</td>\n",
       "      <td>8.609579</td>\n",
       "      <td>70.506086</td>\n",
       "      <td>29.493916</td>\n",
       "      <td>-0.000002</td>\n",
       "      <td>0.404121</td>\n",
       "      <td>0.169051</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                    sample    Sample_Type Primary_Site   ATG7  ATG7_1  ATG7_2  \\\n",
       "0  GTEX-ZTTD-0326-SM-57WFW  Normal Tissue       Muscle  5.042   3.860   2.390   \n",
       "1  GTEX-PX3G-1626-SM-2S1PT  Normal Tissue       Muscle  4.252   2.322   1.956   \n",
       "2  GTEX-OXRO-1726-SM-3LK6C  Normal Tissue       Muscle  3.909   2.602   1.345   \n",
       "\n",
       "   ATG7_3  ATG7(total)    ATG7(1)   ATG7(2)       ATG7(3)  \\\n",
       "0  0.2642    32.944283  14.519306  5.240574  1.199970e+00   \n",
       "1  1.1320    19.052710   4.999249  3.878848  2.190624e+00   \n",
       "2 -9.9660    15.020948   6.070277  2.539302 -1.495113e-07   \n",
       "\n",
       "   log2_+1_ATG7(total)  log2_+1_ATG7(1)  log2_+1_ATG7(2)  ATG7_prot_tot  \\\n",
       "0             5.085097         3.955992         2.641679      20.959850   \n",
       "1             4.325725         2.584782         2.286540      11.068720   \n",
       "2             4.001888         2.821767         1.823465       8.609579   \n",
       "\n",
       "     ATG7_1%    ATG7_2%    ATG7_3%  ATG7(1)norm  ATG7(2)norm  \n",
       "0  69.271996  25.002916   5.725088     0.440723     0.159074  \n",
       "1  45.165557  35.043325  19.791118     0.262390     0.203585  \n",
       "2  70.506086  29.493916  -0.000002     0.404121     0.169051  "
      ]
     },
     "execution_count": 11,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# df with only Normal and Primary \n",
    "df_Norm_Prim = pandas.concat([df_ATG7_Normal, df_ATG7_Primary], axis=0)\n",
    "df_Norm_Prim.reset_index(drop=True, inplace=True)\n",
    "df_Norm_Prim.head(3)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "CGVz9_mqWgpw"
   },
   "source": [
    "### High/Low expression per Median for Isoform expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {
    "id": "Ypc3K2bKWzpI"
   },
   "outputs": [],
   "source": [
    "# What is the median for iso 1 and iso 2?\n",
    "#High >50% and low <50%"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {
    "id": "oA7fE5E3pxi_"
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:11: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  # This is added back by InteractiveShellApp.init_path()\n",
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:14: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  \n"
     ]
    }
   ],
   "source": [
    "Data = [[df_ATG7_Normal, 'Normal'], [df_ATG7_Primary, 'Primary']]\n",
    "isoforms = [[\"ATG7_1\", \"iso1\"], [\"ATG7_2\", \"iso2\"]]\n",
    "\n",
    "df_LoHi = [] \n",
    "\n",
    "for df in Data:\n",
    "  for isoform in isoforms:\n",
    "    xMed = df[0][isoform[0]].median()\n",
    "\n",
    "    low = df[0].loc[df[0][isoform[0]] < xMed]\n",
    "    low['L/H'] = \"Low_\"+isoform[1]\n",
    "\n",
    "    high = df[0].loc[df[0][isoform[0]] > xMed]\n",
    "    high['L/H'] = \"High_\"+isoform[1]\n",
    "\n",
    "  #merge low and high\n",
    "  concat = pandas.concat([low, high], axis=0)\n",
    "  col = concat.pop('L/H')\n",
    "  concat.insert(3, 'L/H', col)\n",
    "  df_LoHi.append(concat)\n",
    "\n",
    "normal_LoHi = df_LoHi[0]\n",
    "primary_LoHi = df_LoHi[1]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "fbjL51C8RltI"
   },
   "source": [
    "## B) Whole gene expression"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 128
    },
    "id": "dpYmq2ernzq8",
    "outputId": "1eee1a21-ef99-444b-b365-542063698ae4"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "22796\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>hgnc_symbol</th>\n",
       "      <th>entrezgene_id</th>\n",
       "      <th>transcript_biotype</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000198888</td>\n",
       "      <td>MT-ND1</td>\n",
       "      <td>4535.0</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000198763</td>\n",
       "      <td>MT-ND2</td>\n",
       "      <td>4536.0</td>\n",
       "      <td>protein_coding</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   ensembl_gene_id hgnc_symbol  entrezgene_id transcript_biotype\n",
       "1  ENSG00000198888      MT-ND1         4535.0     protein_coding\n",
       "2  ENSG00000198763      MT-ND2         4536.0     protein_coding"
      ]
     },
     "execution_count": 14,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#Import list of protein coding transcripts\n",
    "protein_coding_directory = input_file_directory + \"protein_coding_list.txt\"\n",
    "protein_coding_list= pandas.read_csv(protein_coding_directory, sep = \"\\t\")\n",
    "print(len(protein_coding_list))\n",
    "protein_coding_list.head(2)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {
    "id": "BbKEKGZ2EMRE"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "CPU times: user 2min 26s, sys: 3.53 s, total: 2min 29s\n",
      "Wall time: 2min 29s\n"
     ]
    }
   ],
   "source": [
    "%%time\n",
    "\n",
    "df_gexpr_normal = pandas.read_csv(normal_tissue_expression_file, sep = \"\\t\")\n",
    "df_gexpr_primary = pandas.read_csv(primary_tumor_expression_file, sep = \"\\t\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {
    "id": "oWVf4gAGEMRG"
   },
   "outputs": [],
   "source": [
    "#Preparation protein coding gene expression\n",
    "#remove dot in ENSEMBL name. Function split created at the beginning\n",
    "data = [df_gexpr_normal, df_gexpr_primary]\n",
    "\n",
    "df_protcoding_exp = []\n",
    "for df in data: \n",
    "  #remove dot in ens name\n",
    "  df[\"sample\"] = df[\"sample\"].apply(split)\n",
    "  #see if have duplicate\n",
    "  # print(df.duplicated(subset=['sample']).sum())\n",
    "\n",
    "  #Merge with prot coding list\n",
    "  df.rename(columns = {'sample' : 'ensembl_gene_id'}, inplace = True)\n",
    "  dfmerged = protein_coding_list.merge(df)\n",
    "  df_protcoding_exp.append(dfmerged)\n",
    "\n",
    "  #control merge is correct\n",
    "  # print(\"control merge is correct\")\n",
    "  # print(df.loc[df['ensembl_gene_id'] == 'ENSG00000198712'])\n",
    "  # print(dfmerged.loc[dfmerged['ensembl_gene_id'] == 'ENSG00000198712'])\n",
    "\n",
    "normal_protcoding = df_protcoding_exp[0]\n",
    "primary_protcoding = df_protcoding_exp[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "9b0Bg43gEMRM",
    "outputId": "464c1f55-7b1e-44d8-ac3c-3a73ee31bde9"
   },
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_isoform is:  16614\n",
      "df_gene_exp is:  2\n",
      "df_merged:  7429\n"
     ]
    },
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/adrian/anaconda3/lib/python3.7/site-packages/ipykernel_launcher.py:29: FutureWarning: In a future version of pandas all arguments of DataFrame.drop except for the argument 'labels' will be keyword-only\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "df_isoform is:  16614\n",
      "df_gene_exp is:  2\n",
      "df_merged:  9185\n"
     ]
    }
   ],
   "source": [
    "#Preparation of the global dataframe\n",
    "data = [[normal_protcoding, df_ATG7_Normal], [primary_protcoding, df_ATG7_Primary]]\n",
    "\n",
    "df_final = []\n",
    "\n",
    "for df in data:\n",
    "  \n",
    "  #transpose the table \n",
    "  curr_df = df[0].copy()\n",
    "  curr_df = numpy.transpose(curr_df)\n",
    "\n",
    "  #Change head column & delete the name of the column index; choose ENSEMBL\n",
    "  curr_df.columns = curr_df.iloc[0]\n",
    "  curr_df.columns.name = None\n",
    "  curr_df = curr_df.reset_index()\n",
    "\n",
    "  #Delete the first rows & columns rename to merge\n",
    "  curr_df = curr_df.drop(curr_df.index[0:4])\n",
    "  curr_df.rename(columns = {'index' : 'sample'}, inplace = True)\n",
    "  \n",
    "  #Merge the two df to have isoform expression + whole gene expression \n",
    "  df_merged = pandas.merge(df[1], curr_df, on= 'sample')\n",
    "\n",
    "  print('df_isoform is: ', len(df_Norm_Prim))\n",
    "  print('df_gene_exp is: ', len(df))\n",
    "  print('df_merged: ', len(df_merged))\n",
    "\n",
    "  #delete the sample column and select column\n",
    "  df_merged = df_merged.drop(\"sample\", 1)\n",
    "  df_merged = df_merged.drop(df_merged.columns[5:19], axis=1)\n",
    "  \n",
    "  #save in list\n",
    "  df_final.append(df_merged)\n",
    "\n",
    "\n",
    "df_normal_expr = df_final[0]\n",
    "df_primary_expr = df_final[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {
    "id": "s_4JIRYMnuti"
   },
   "outputs": [
    {
     "ename": "KeyError",
     "evalue": "'sample'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3360\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3361\u001b[0;31m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3362\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/_libs/index.pyx\u001b[0m in \u001b[0;36mpandas._libs.index.IndexEngine.get_loc\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;32mpandas/_libs/hashtable_class_helper.pxi\u001b[0m in \u001b[0;36mpandas._libs.hashtable.PyObjectHashTable.get_item\u001b[0;34m()\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'sample'",
      "\nThe above exception was the direct cause of the following exception:\n",
      "\u001b[0;31mKeyError\u001b[0m                                  Traceback (most recent call last)",
      "\u001b[0;32m<ipython-input-18-926aae495e80>\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0;32mfor\u001b[0m \u001b[0mdf\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      7\u001b[0m   \u001b[0;31m#remove dot in ens name\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 8\u001b[0;31m   \u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"sample\"\u001b[0m\u001b[0;34m]\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mdf\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m\"sample\"\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mapply\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msplit\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      9\u001b[0m   \u001b[0;31m#see if have duplicate\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     10\u001b[0m   \u001b[0;31m# print(df.duplicated(subset=['sample']).sum())\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/frame.py\u001b[0m in \u001b[0;36m__getitem__\u001b[0;34m(self, key)\u001b[0m\n\u001b[1;32m   3456\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mnlevels\u001b[0m \u001b[0;34m>\u001b[0m \u001b[0;36m1\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3457\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_getitem_multilevel\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3458\u001b[0;31m             \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mcolumns\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3459\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mis_integer\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3460\u001b[0m                 \u001b[0mindexer\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34m[\u001b[0m\u001b[0mindexer\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/anaconda3/lib/python3.7/site-packages/pandas/core/indexes/base.py\u001b[0m in \u001b[0;36mget_loc\u001b[0;34m(self, key, method, tolerance)\u001b[0m\n\u001b[1;32m   3361\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_engine\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_loc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mcasted_key\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3362\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mKeyError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 3363\u001b[0;31m                 \u001b[0;32mraise\u001b[0m \u001b[0mKeyError\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   3364\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   3365\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mis_scalar\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0misna\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mkey\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mand\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhasnans\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyError\u001b[0m: 'sample'"
     ]
    }
   ],
   "source": [
    "#Preparation protein coding gene expression\n",
    "#remove dot in ENSEMBL name. Function split created at the beginning\n",
    "data = [df_gexpr_normal, df_gexpr_primary]\n",
    "\n",
    "df_protcoding_exp = []\n",
    "for df in data: \n",
    "  #remove dot in ens name\n",
    "  df[\"sample\"] = df[\"sample\"].apply(split)\n",
    "  #see if have duplicate\n",
    "  # print(df.duplicated(subset=['sample']).sum())\n",
    "\n",
    "  #Merge with prot coding list\n",
    "  df.rename(columns = {'sample' : 'ensembl_gene_id'}, inplace = True)\n",
    "  dfmerged = protein_coding_list.merge(df)\n",
    "  df_protcoding_exp.append(dfmerged)\n",
    "\n",
    "  #control merge is correct\n",
    "  # print(\"control merge is correct\")\n",
    "  # print(df.loc[df['ensembl_gene_id'] == 'ENSG00000198712'])\n",
    "  # print(dfmerged.loc[dfmerged['ensembl_gene_id'] == 'ENSG00000198712'])\n",
    "\n",
    "normal_protcoding = df_protcoding_exp[0]\n",
    "primary_protcoding = df_protcoding_exp[1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>GTEX-1117F-0226-SM-5GZZ7</th>\n",
       "      <th>GTEX-1117F-0426-SM-5EGHI</th>\n",
       "      <th>GTEX-1117F-0526-SM-5EGHJ</th>\n",
       "      <th>GTEX-1117F-0626-SM-5N9CS</th>\n",
       "      <th>GTEX-1117F-0726-SM-5GIEN</th>\n",
       "      <th>GTEX-1117F-1326-SM-5EGHH</th>\n",
       "      <th>GTEX-1117F-2226-SM-5N9CH</th>\n",
       "      <th>GTEX-1117F-2426-SM-5EGGH</th>\n",
       "      <th>GTEX-1117F-2826-SM-5GZXL</th>\n",
       "      <th>...</th>\n",
       "      <th>GTEX-ZZPU-0726-SM-5N9C8</th>\n",
       "      <th>GTEX-ZZPU-0926-SM-5GZYT</th>\n",
       "      <th>GTEX-ZZPU-1226-SM-5N9CK</th>\n",
       "      <th>GTEX-ZZPU-1326-SM-5GZWS</th>\n",
       "      <th>GTEX-ZZPU-1426-SM-5GZZ6</th>\n",
       "      <th>GTEX-ZZPU-1826-SM-5E43L</th>\n",
       "      <th>GTEX-ZZPU-2126-SM-5EGIU</th>\n",
       "      <th>GTEX-ZZPU-2226-SM-5EGIV</th>\n",
       "      <th>GTEX-ZZPU-2626-SM-5E45Y</th>\n",
       "      <th>GTEX-ZZPU-2726-SM-5NQ8O</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000242268</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-1.2481</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>-1.7809</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>...</td>\n",
       "      <td>-0.8084</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000259041</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000270112</td>\n",
       "      <td>-4.2934</td>\n",
       "      <td>0.0014</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-5.5735</td>\n",
       "      <td>0.3573</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-6.5064</td>\n",
       "      <td>-5.0116</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-1.8314</td>\n",
       "      <td>-1.7809</td>\n",
       "      <td>-2.0529</td>\n",
       "      <td>-6.5064</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-2.8262</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000167578</td>\n",
       "      <td>5.1190</td>\n",
       "      <td>4.1277</td>\n",
       "      <td>4.4067</td>\n",
       "      <td>5.6860</td>\n",
       "      <td>4.0357</td>\n",
       "      <td>4.6849</td>\n",
       "      <td>4.5009</td>\n",
       "      <td>5.3954</td>\n",
       "      <td>4.9402</td>\n",
       "      <td>...</td>\n",
       "      <td>3.9099</td>\n",
       "      <td>3.3478</td>\n",
       "      <td>5.5201</td>\n",
       "      <td>5.6930</td>\n",
       "      <td>4.2072</td>\n",
       "      <td>4.2788</td>\n",
       "      <td>4.0790</td>\n",
       "      <td>4.7856</td>\n",
       "      <td>3.5387</td>\n",
       "      <td>4.7071</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000278814</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 7430 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ensembl_gene_id  GTEX-1117F-0226-SM-5GZZ7  GTEX-1117F-0426-SM-5EGHI  \\\n",
       "0  ENSG00000242268                   -9.9658                   -9.9658   \n",
       "1  ENSG00000259041                   -9.9658                   -9.9658   \n",
       "2  ENSG00000270112                   -4.2934                    0.0014   \n",
       "3  ENSG00000167578                    5.1190                    4.1277   \n",
       "4  ENSG00000278814                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-0526-SM-5EGHJ  GTEX-1117F-0626-SM-5N9CS  \\\n",
       "0                   -9.9658                   -1.2481   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                   -9.9658                   -5.5735   \n",
       "3                    4.4067                    5.6860   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-0726-SM-5GIEN  GTEX-1117F-1326-SM-5EGHH  \\\n",
       "0                   -3.8160                   -1.7809   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                    0.3573                   -9.9658   \n",
       "3                    4.0357                    4.6849   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-2226-SM-5N9CH  GTEX-1117F-2426-SM-5EGGH  \\\n",
       "0                   -9.9658                   -9.9658   \n",
       "1                   -9.9658                   -9.9658   \n",
       "2                   -6.5064                   -5.0116   \n",
       "3                    4.5009                    5.3954   \n",
       "4                   -9.9658                   -9.9658   \n",
       "\n",
       "   GTEX-1117F-2826-SM-5GZXL  ...  GTEX-ZZPU-0726-SM-5N9C8  \\\n",
       "0                   -3.6259  ...                  -0.8084   \n",
       "1                   -9.9658  ...                  -9.9658   \n",
       "2                   -9.9658  ...                  -4.0350   \n",
       "3                    4.9402  ...                   3.9099   \n",
       "4                   -9.9658  ...                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-0926-SM-5GZYT  GTEX-ZZPU-1226-SM-5N9CK  GTEX-ZZPU-1326-SM-5GZWS  \\\n",
       "0                  -9.9658                  -9.9658                  -9.9658   \n",
       "1                  -9.9658                  -9.9658                  -9.9658   \n",
       "2                  -1.8314                  -1.7809                  -2.0529   \n",
       "3                   3.3478                   5.5201                   5.6930   \n",
       "4                  -9.9658                  -9.9658                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-1426-SM-5GZZ6  GTEX-ZZPU-1826-SM-5E43L  GTEX-ZZPU-2126-SM-5EGIU  \\\n",
       "0                  -4.0350                  -9.9658                  -9.9658   \n",
       "1                  -9.9658                  -9.9658                  -9.9658   \n",
       "2                  -6.5064                  -3.6259                  -9.9658   \n",
       "3                   4.2072                   4.2788                   4.0790   \n",
       "4                  -9.9658                  -9.9658                  -9.9658   \n",
       "\n",
       "   GTEX-ZZPU-2226-SM-5EGIV  GTEX-ZZPU-2626-SM-5E45Y  GTEX-ZZPU-2726-SM-5NQ8O  \n",
       "0                  -9.9658                  -9.9658                  -3.6259  \n",
       "1                  -9.9658                  -9.9658                  -9.9658  \n",
       "2                  -9.9658                  -2.8262                  -9.9658  \n",
       "3                   4.7856                   3.5387                   4.7071  \n",
       "4                  -9.9658                  -9.9658                  -9.9658  \n",
       "\n",
       "[5 rows x 7430 columns]"
      ]
     },
     "execution_count": 20,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gexpr_normal.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>ensembl_gene_id</th>\n",
       "      <th>TCGA-02-0047-01</th>\n",
       "      <th>TCGA-02-0055-01</th>\n",
       "      <th>TCGA-02-2483-01</th>\n",
       "      <th>TCGA-02-2485-01</th>\n",
       "      <th>TCGA-04-1331-01</th>\n",
       "      <th>TCGA-04-1332-01</th>\n",
       "      <th>TCGA-04-1337-01</th>\n",
       "      <th>TCGA-04-1338-01</th>\n",
       "      <th>TCGA-04-1341-01</th>\n",
       "      <th>...</th>\n",
       "      <th>TCGA-ZP-A9D4-01</th>\n",
       "      <th>TCGA-ZQ-A9CR-01</th>\n",
       "      <th>TCGA-ZR-A9CJ-01</th>\n",
       "      <th>TCGA-ZS-A9CD-01</th>\n",
       "      <th>TCGA-ZS-A9CE-01</th>\n",
       "      <th>TCGA-ZS-A9CF-01</th>\n",
       "      <th>TCGA-ZS-A9CG-01</th>\n",
       "      <th>TCGA-ZT-A8OM-01</th>\n",
       "      <th>TCGA-ZU-A8S4-01</th>\n",
       "      <th>TCGA-ZX-AA5X-01</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>ENSG00000242268</td>\n",
       "      <td>0.0158</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-0.6193</td>\n",
       "      <td>-5.0116</td>\n",
       "      <td>-3.4580</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-4.6082</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-3.6259</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>ENSG00000259041</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>ENSG00000270112</td>\n",
       "      <td>-0.8863</td>\n",
       "      <td>-3.8160</td>\n",
       "      <td>-5.5735</td>\n",
       "      <td>-1.5522</td>\n",
       "      <td>0.4761</td>\n",
       "      <td>-2.4659</td>\n",
       "      <td>-5.0116</td>\n",
       "      <td>-4.0350</td>\n",
       "      <td>-2.7274</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-6.5064</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>ENSG00000167578</td>\n",
       "      <td>5.9750</td>\n",
       "      <td>5.2506</td>\n",
       "      <td>5.4519</td>\n",
       "      <td>6.5390</td>\n",
       "      <td>4.9514</td>\n",
       "      <td>6.4481</td>\n",
       "      <td>5.1595</td>\n",
       "      <td>4.0506</td>\n",
       "      <td>4.5249</td>\n",
       "      <td>...</td>\n",
       "      <td>3.6136</td>\n",
       "      <td>4.4700</td>\n",
       "      <td>4.1780</td>\n",
       "      <td>4.5547</td>\n",
       "      <td>3.6737</td>\n",
       "      <td>4.9331</td>\n",
       "      <td>3.7646</td>\n",
       "      <td>5.5201</td>\n",
       "      <td>5.4216</td>\n",
       "      <td>4.7991</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>ENSG00000278814</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>...</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "      <td>-9.9658</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 9186 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "   ensembl_gene_id  TCGA-02-0047-01  TCGA-02-0055-01  TCGA-02-2483-01  \\\n",
       "0  ENSG00000242268           0.0158          -9.9658          -9.9658   \n",
       "1  ENSG00000259041          -9.9658          -9.9658          -9.9658   \n",
       "2  ENSG00000270112          -0.8863          -3.8160          -5.5735   \n",
       "3  ENSG00000167578           5.9750           5.2506           5.4519   \n",
       "4  ENSG00000278814          -9.9658          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-02-2485-01  TCGA-04-1331-01  TCGA-04-1332-01  TCGA-04-1337-01  \\\n",
       "0          -0.6193          -5.0116          -3.4580          -3.8160   \n",
       "1          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "2          -1.5522           0.4761          -2.4659          -5.0116   \n",
       "3           6.5390           4.9514           6.4481           5.1595   \n",
       "4          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-04-1338-01  TCGA-04-1341-01  ...  TCGA-ZP-A9D4-01  TCGA-ZQ-A9CR-01  \\\n",
       "0          -9.9658          -9.9658  ...          -9.9658          -9.9658   \n",
       "1          -9.9658          -9.9658  ...          -9.9658          -9.9658   \n",
       "2          -4.0350          -2.7274  ...          -9.9658          -9.9658   \n",
       "3           4.0506           4.5249  ...           3.6136           4.4700   \n",
       "4          -9.9658          -9.9658  ...          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-ZR-A9CJ-01  TCGA-ZS-A9CD-01  TCGA-ZS-A9CE-01  TCGA-ZS-A9CF-01  \\\n",
       "0          -9.9658          -4.6082          -9.9658          -9.9658   \n",
       "1          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "2          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "3           4.1780           4.5547           3.6737           4.9331   \n",
       "4          -9.9658          -9.9658          -9.9658          -9.9658   \n",
       "\n",
       "   TCGA-ZS-A9CG-01  TCGA-ZT-A8OM-01  TCGA-ZU-A8S4-01  TCGA-ZX-AA5X-01  \n",
       "0          -9.9658          -3.6259          -9.9658          -9.9658  \n",
       "1          -9.9658          -9.9658          -9.9658          -9.9658  \n",
       "2          -9.9658          -9.9658          -9.9658          -6.5064  \n",
       "3           3.7646           5.5201           5.4216           4.7991  \n",
       "4          -9.9658          -9.9658          -9.9658          -9.9658  \n",
       "\n",
       "[5 rows x 9186 columns]"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df_gexpr_primary.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "vXuPgJcIg916"
   },
   "outputs": [],
   "source": [
    "#Preparation of the global dataframe\n",
    "# data = [[df_protcoding[0], 'normal_protcoding'], [df_protcoding[1], 'primary_protcoding']]\n",
    "data = [[Normal, normal_protcoding], [Primary, primary_protcoding]]\n",
    "\n",
    "df_final = []\n",
    "\n",
    "for df in data:\n",
    "  \n",
    "  #transpose the table \n",
    "  curr_df = df[0].copy()\n",
    "  curr_df = numpy.transpose(curr_df)\n",
    "\n",
    "  #Change head column & delete the name of the column index; choose ENSEMBL\n",
    "  curr_df.columns = curr_df.iloc[0]\n",
    "  curr_df.columns.name = None\n",
    "  curr_df = curr_df.reset_index()\n",
    "\n",
    "  #Delete the first rows & columns rename to merge\n",
    "  curr_df = curr_df.drop(curr_df.index[0:4])\n",
    "  curr_df.rename(columns = {'index' : 'sample'}, inplace = True)\n",
    "\n",
    "  #Merge the two df to have isoform expression + whole gene expression \n",
    "  df_merged = pandas.merge(df[1], curr_df, on= 'sample')\n",
    "\n",
    "  print('df_isoform is: ', len(df_Norm_Prim))\n",
    "  print('df_gene_exp is: ', len(df))\n",
    "  print('df_merged: ', len(df_merged))\n",
    "\n",
    "  #delete the sample column and select column\n",
    "  df_merged = df_merged.drop(\"sample\", 1)\n",
    "  df_merged = gexpr_compl.drop(gexpr_compl.columns[5:19], axis=1)\n",
    "  print(\"controler si j'ai bien ATG7_2\", 'suivi de ENSG00000198888')\n",
    "  \n",
    "  #save in list\n",
    "  df_final.append(df_merged)\n",
    "\n",
    "\n",
    "df_final[0] = df_normal_expr\n",
    "df_final[1] = df_primary_expr"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "R8fLWHXOBYJl"
   },
   "source": [
    "# III] ATG7 expression"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ufG3BGqPkmrk"
   },
   "source": [
    "##A) Distribution\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "vl1-Yk-8NVZi",
    "outputId": "227593fd-74c8-4670-8ad4-8a0ac27f1752"
   },
   "outputs": [],
   "source": [
    "# Distribution graph, for ATG7, iso1 & iso2\n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'in all tissue', df_ATG7_Primary, 'all primary tumors']]\n",
    "\n",
    "isoforms = [['log2_+1_ATG7(total)', 'ATG7'], ['log2_+1_ATG7(1)', 'ATG7(1)'], ['log2_+1_ATG7(2)', 'ATG7(2)']]\n",
    "\n",
    "\n",
    "for df in all_dfs:\n",
    "  for isoform in isoforms:\n",
    "    xData = df[0][isoform[0]]\n",
    "    x2Data = df[2][isoform[0]]\n",
    "\n",
    "    seaborn.kdeplot(xData, color = 'mediumaquamarine', fill = True)\n",
    "    seaborn.kdeplot(x2Data, color = 'sandybrown', fill = True)\n",
    "\n",
    "    plt.xlabel('log2 '+ isoform[1]+ ' +1')\n",
    "    plt.title(isoform[1]+ ' expression ' + df[1] )\n",
    "    plt.legend(labels=[\"Normal Tissue\",\"Primary Tumor \"])\n",
    "    plt.show()\n",
    "    plt.clf()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "PL_4aao-49o9"
   },
   "source": [
    "##B) Percentage do not express ATG7 isoforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "zwxExNKnmvhu",
    "outputId": "79575762-1bd2-4c3e-c35d-ede5886f24c6"
   },
   "outputs": [],
   "source": [
    "#Percentage of patients: \n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'all normal tissue'], [df_ATG7_Primary, 'all primary tumor']]\n",
    "\n",
    "isoforms = [['ATG7(total)', 'ATG7'], ['ATG7(1)', 'ATG7(1)'], ['ATG7(2)', 'ATG7(2)'], ['ATG7(3)', 'ATG7(3)']]\n",
    "\n",
    "for df in all_dfs:\n",
    "  print(df[1])\n",
    "  for isoform in isoforms:\n",
    "        \n",
    "    QUOI = isoform[1]\n",
    "    OU = df[1]\n",
    "    all = (df[0][isoform[0]]).shape\n",
    "    no_exp = (df[0][isoform[0]]<0.1).sum()\n",
    "    perc_no_exp = (no_exp*100)/all\n",
    "    perc_no_exp = float(perc_no_exp)\n",
    "\n",
    "    # print(\"The percentage of patient who do not express\", QUOI, OU, \"is:\", round(perc_no_exp,1),'%')\n",
    "    print(round(perc_no_exp,1),'% of tumor do not express', QUOI)\n",
    "  print()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "l9R88UMYsp9h"
   },
   "source": [
    "## C) Expression figures"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "j87nxfPMrMLB"
   },
   "source": [
    "### 1) Value TPM"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 569
    },
    "id": "3_zvT1lweuYb",
    "outputId": "d561789f-e155-4e1d-ba5a-ac1a107c389a"
   },
   "outputs": [],
   "source": [
    "# Distribution: Data preparation + graph\n",
    "\n",
    "data = [[df_Norm_Prim, 'all tissue']]\n",
    "\n",
    "for df in data:\n",
    "   \n",
    "  shape = df[0].shape\n",
    "  shape = shape[0]\n",
    "  print(shape)\n",
    "\n",
    "  #select the sample type column\n",
    "  sample_type = {'sample_type':[df[0].iloc[:,1][i]\n",
    "      for i in range(0, shape)]}\n",
    "\n",
    "  #convert to df, and copy it 3 times\n",
    "  sample_typedf = pandas.DataFrame(sample_type)\n",
    "  sample_typedf = pandas.concat([sample_typedf,sample_typedf,sample_typedf])\n",
    "  sample_typedf.reset_index(drop=True, inplace=True)\n",
    "\n",
    "  # Y for expression value \n",
    "  y = {'y':[df[0].iloc[:,j][i]\n",
    "      for j in range(9,12)\n",
    "        for i in range(0, shape)]}\n",
    "\n",
    "  #create df with Y \n",
    "  data_distrib = pandas.DataFrame(y)\n",
    "\n",
    "  # X for ATG7/iso1/iso2\n",
    "  data_distrib['x'] = ''\n",
    "  data_distrib['x'][: shape] = 'ATG7'\n",
    "  data_distrib['x'][shape: shape*2] = 'ATG7(1)'\n",
    "  data_distrib['x'][shape*2:] = 'ATG7(2)'\n",
    "\n",
    "  # Will be used for Hue\n",
    "  data_distrib['Sample Type'] = sample_typedf['sample_type']\n",
    "\n",
    "  #graph\n",
    "  seaborn.catplot(x=\"x\", y=\"y\", hue=\"Sample Type\", aspect= 1.2, kind=\"box\", palette = 'Set2', data = data_distrib, \n",
    "                  boxprops={'lw':2}, medianprops={'lw':2}, whiskerprops={'lw':2}, showcaps=True, showfliers=False)\n",
    "  \n",
    "  plt.ylabel('Expression of ATG7 (TPM)')\n",
    "  plt.xlabel('')\n",
    "  plt.title('ATG7 expression in '+ df[1], x=.55)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "lqjTE5owk0cD",
    "outputId": "88756d11-ca02-4610-96f5-e52fc09ae047"
   },
   "outputs": [],
   "source": [
    "#Distribution: Stats for all \n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'all normal tissue', df_ATG7_Primary, 'all primary tumors']]\n",
    "\n",
    "isoforms = ['ATG7', 'ATG7_1', 'ATG7_2']\n",
    "\n",
    "for df in all_dfs:\n",
    "  for isoform in isoforms:\n",
    "\n",
    "    n1 = \"{:.3f}\".format(numpy.median(df[0][isoform]))\n",
    "    n2 = \"{:.3f}\".format(numpy.median(df[2][isoform]))\n",
    "\n",
    "    # Mann-Whitney analysis \n",
    "    statistic, pvalue = scipy.stats.mannwhitneyu(df[0][isoform], df[2][isoform])\n",
    "    Pvalue = \"{:.3E}\".format(pvalue)\n",
    "    print(\"the pvalue between\", df[1]+'('+n1+')', \"and\", df[3]+'('+n2+')', \"for\", isoform, 'is', Pvalue)\n",
    "  print()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "rccgpnCWrUyE"
   },
   "source": [
    "### 2) Normalized"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 722
    },
    "id": "n_h00in7pPDk",
    "outputId": "90718d06-95e5-4f40-8c1e-e18cb5a259a3"
   },
   "outputs": [],
   "source": [
    "# NORMALiZED Distribution: Data preparation + graph\n",
    "\n",
    "data = [[df_Norm_Prim, 'all_tissue']]\n",
    "\n",
    "\n",
    "for df in data:\n",
    "   \n",
    "  shape = df[0].shape\n",
    "  shape = shape[0]\n",
    "  print(shape)\n",
    "\n",
    "  #select the sample type column\n",
    "  sample_type = {'sample_type':[df[0].iloc[:,1][i]\n",
    "      for i in range(0, shape)]}\n",
    "\n",
    "  #convert to df, and copy it 3 times\n",
    "  sample_typedf = pandas.DataFrame(sample_type)\n",
    "  sample_typedf = pandas.concat([sample_typedf,sample_typedf,sample_typedf])\n",
    "  sample_typedf.reset_index(drop=True, inplace=True)\n",
    "\n",
    "  # Y for expression value \n",
    "  y = {'y':[df[0].iloc[:,j][i]\n",
    "      for j in range(18,20)\n",
    "        for i in range(0, shape)]}\n",
    "\n",
    "  #create df with Y \n",
    "  data_distrib = pandas.DataFrame(y)\n",
    "\n",
    "  # X for iso1/ATG7 and iso2/ATG7\n",
    "  data_distrib['x'] = ''\n",
    "  data_distrib['x'][: shape] = 'ATG7(1)/ATG7'\n",
    "  data_distrib['x'][shape:] = 'ATG7(2)/ATG7'\n",
    "\n",
    "\n",
    "  # Will be used for Hue\n",
    "  data_distrib['Sample Type'] = sample_typedf['sample_type']\n",
    "\n",
    "  #graph\n",
    "  seaborn.catplot(x=\"x\", y=\"y\", hue=\"Sample Type\", aspect= 1.2, kind=\"box\", palette = 'Set2', data = data_distrib, \n",
    "                  boxprops={'lw':2}, medianprops={'lw':2}, whiskerprops={'lw':2}, showcaps=True, showfliers=False)\n",
    "  \n",
    "  plt.ylabel('ATG7 isoforms/total ATG7 in TPM')\n",
    "  plt.xlabel('')\n",
    "  plt.title('ATG7 proportion in '+ df[1], x=0.55)\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/"
    },
    "id": "JflZDsJbpPDl",
    "outputId": "064d15f1-d035-4b7b-f4d5-bf8e473a8518"
   },
   "outputs": [],
   "source": [
    "#NORMALIZED Distribution: Stats for all \n",
    "\n",
    "all_dfs = [[df_ATG7_Normal, 'all normal tissue', df_ATG7_Primary, 'all primary tumors']]\n",
    "\n",
    "isoforms = ['ATG7(1)norm', 'ATG7(2)norm']\n",
    "print('NORMALIZED')\n",
    "\n",
    "for df in all_dfs:\n",
    "  for isoform in isoforms:\n",
    "\n",
    "    n1 = \"{:.3f}\".format(numpy.median(df[0][isoform]))\n",
    "    n2 = \"{:.3f}\".format(numpy.median(df[2][isoform]))\n",
    "\n",
    "    # Mann-Whitney analysis \n",
    "    statistic, pvalue = scipy.stats.mannwhitneyu(df[0][isoform], df[2][isoform])\n",
    "    Pvalue = \"{:.3E}\".format(pvalue)\n",
    "    print(\"the pvalue between\", df[1]+'('+n1+')', \"and\", df[3]+'('+n2+')', \"for\", isoform, 'is', Pvalue)\n",
    "  print()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "ivLjI2vHTV5A"
   },
   "source": [
    "#IV] Expression correlation "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "1Pspjsp_10nE"
   },
   "outputs": [],
   "source": [
    "#To see information. \n",
    "# → I can see many samples have no expression of ATG7(2) so I loose many data in normal tissue when removing <4tpm.\n",
    "#If I want to work on Normal, I have to do exeption on iso2\n",
    "\n",
    "all_tissue_used = [[df_normal_exp,'Normal Tissue'], [df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "isoforms = ['ATG7_1', 'ATG7_2']\n",
    "\n",
    "for tissue in all_tissue_used:\n",
    "  print(tissue[1])\n",
    "  ff = tissue[0].shape\n",
    "  print('number of patients:', ff[0])\n",
    "  print(\"ATG7_1 max is\", tissue[0]['ATG7_1'].max())\n",
    "  print(\"ATG7_2 max is\", tissue[0]['ATG7_2'].max())\n",
    "    \n",
    "\n",
    "  for isoform in isoforms:\n",
    "    print('shape before removing', isoform, 'zero expression', tissue[0].shape)\n",
    "    df_atleast1 = tissue[0][tissue[0][isoform]>=0]\n",
    "    print('shape after removing', isoform, 'zero expression', df_atleast1.shape)\n",
    "\n",
    "    #create a new row with max() for each gene\n",
    "    tissue[0].loc['max()'] = tissue[0].max()\n",
    "\n",
    "    #Transpose to have max() in column\n",
    "    tissue[0] = tissue[0].T\n",
    "    nb_before = tissue[0].shape\n",
    "    tissue[0] = tissue[0][tissue[0]['max()'] > 1.2]\n",
    "    nb_after = tissue[0].shape\n",
    "    print(\"number of genes removed:\", nb_before[0]-nb_after[0])\n",
    "    print('now the minimum of TPM is', tissue[0]['max()'].min())\n",
    "\n",
    "    #delete the column max()\n",
    "    tissue[0] =  tissue[0].drop('max()', axis = 1) \n",
    "    # print(tissue[0].shape)\n",
    "\n",
    "    #transpose back\n",
    "    tissue[0] = tissue[0].T\n",
    "\n",
    "\n",
    "    print()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "Et5DpsPy4DBP"
   },
   "source": [
    "##1) between the two isoforms"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "colab": {
     "base_uri": "https://localhost:8080/",
     "height": 1000
    },
    "id": "4azGSbHX1l11",
    "outputId": "8e6361eb-8c82-479d-c71e-2bf6fb6c743e"
   },
   "outputs": [],
   "source": [
    "#isoform correlation in all tissue \n",
    "\n",
    "#three values are very high and make my graph ugly... I remove them.\n",
    "df_ATG7_Primary2 = Primary[Primary['ATG7(1)'] < 30 ]\n",
    "print(Primary.shape)\n",
    "print(Primary2.shape)\n",
    "\n",
    "\n",
    "tissues = [[df_ATG7_Normal, \"normal tissue\"], [df_ATG7_Primary2, \"primary tumors\"]]\n",
    "\n",
    "for tissue in tissues: \n",
    "  x = tissue[0]['ATG7(1)'].to_list()\n",
    "  y = tissue[0]['ATG7(2)'].to_list()\n",
    "\n",
    "  r_value, p_value = scipy.stats.spearmanr(x, y)\n",
    "  print(r_value, p_value)\n",
    "\n",
    "  seaborn.regplot(x=x,y=y, x_bins=800, x_ci = 0, marker=\"+\")\n",
    "  name = \"Correlation between the two isoforms in \"+  tissue[1]\n",
    "  plt.title(name)\n",
    "  plt.xlabel(\"ATG7(1) expression in tpm\")\n",
    "  plt.ylabel('ATG7(2) expression tpm')\n",
    "  plt.show()\n",
    "  plt.clf()\n"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "id": "RgHYwrKD4LgQ"
   },
   "source": [
    "##1) between all protein coding"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "LxKzIf-63QkM"
   },
   "outputs": [],
   "source": [
    "#to know the ensembl code for HK2\n",
    "protein_coding_list.loc[protein_coding_list['hgnc_symbol'] == 'MFAP3']"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "hsgaqjKl3z5F"
   },
   "outputs": [],
   "source": [
    "# Spearman Correlation, main genes \n",
    "\n",
    "all_tissue_used = [[df_normal_exp,'Normal Tissue'], [df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "HK2 = 'ENSG00000159399'\n",
    "YAP1 = 'ENSG00000137693'\n",
    "AJUBA = 'ENSG00000129474'\n",
    "YBX1 = 'ENSG00000065978'\n",
    "TGFB1 = 'ENSG00000105329'\n",
    "ATG7_1 = 'ATG7_1'\n",
    "ATG7_2 = 'ATG7_2'\n",
    "MTDH = 'ENSG00000147649'\n",
    "MAPK1 = 'ENSG00000100030'\n",
    "KPNA1 = 'ENSG00000114030'\n",
    "TNPO1 = 'ENSG00000083312'\n",
    "\n",
    "for tissue in all_tissue_used:\n",
    "  genes_x = [[KPNA1,'KPNA1'], [MAPK1, 'MAPK1'], [TNPO1,'TNPO1'], [MTDH,'MTDH'], \n",
    "             [HK2, \"HK2\"], [YAP1, 'YAP1'], [AJUBA,'AJUBA'], [YBX1,'YBX1'], \n",
    "             [TGFB1,'TGFB1'], [ATG7_1,'ATG7_1'], [ATG7_2,'ATG7_2']]\n",
    "\n",
    "  print(tissue[1])\n",
    "\n",
    "  for gene in genes_x:\n",
    "    x = tissue[0][gene[0]].to_list()\n",
    "    y = tissue[0]['ATG7'].to_list()\n",
    "    y1 = tissue[0]['ATG7_1'].to_list()\n",
    "    y2 = tissue[0]['ATG7_2'].to_list()  \n",
    "  \n",
    "    r_value, p_value = scipy.stats.spearmanr(x, y)\n",
    "    print('  correlation between ATG7 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "    r_value, p_value = scipy.stats.spearmanr(x, y1)\n",
    "    print('  correlation between ATG7_1 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "    r_value, p_value = scipy.stats.spearmanr(x, y2)\n",
    "    print('  correlation between ATG7_2 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "    print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "DSfvcpoN4r9O"
   },
   "outputs": [],
   "source": [
    "# Spearman Correlation, all genes\n",
    "all_tissue_used = [[df_normal_exp,'Normal Tissues'], [df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "isoforms = ['ATG7_1', 'ATG7_2']\n",
    "\n",
    "all_positiv_corr_spearman = []\n",
    "\n",
    "for tissue in all_tissue_used:\n",
    "  print(tissue[1], tissue[0].shape)\n",
    "\n",
    "  #create a new row with max() for each gene and transpose\n",
    "  tissue[0].loc['max()'] = tissue[0].max()\n",
    "  tissue[0] = tissue[0].T\n",
    "\n",
    "  #select all row where max() > 2.3 (= 5 tpm)\n",
    "  nb_before = tissue[0]['max()'].count()\n",
    "  tissue[0] = tissue[0][tissue[0]['max()'] > 2.3]\n",
    "  nb_after = tissue[0]['max()'].count()\n",
    "  print(\"number of genes removed:\", nb_before-nb_after)\n",
    "  print('the minimum of TPM is', tissue[0]['max()'].min())\n",
    "  \n",
    "  #delete the column max() and tranpose back\n",
    "  tissue[0] =  tissue[0].drop('max()', axis = 1) \n",
    "  tissue[0] = tissue[0].T\n",
    "  print('nombre genes after removing < 4tpm', tissue[0].shape)\n",
    "\n",
    "  for isoform in isoforms:\n",
    "\n",
    "    genepos = []\n",
    "    geneneg = []\n",
    "    rvaluepos = []\n",
    "    rvalueneg = []\n",
    "    pvaluepos = []\n",
    "    pvalueneg = []\n",
    "\n",
    "    print(isoform)\n",
    "    x = tissue[0][isoform].to_list()\n",
    "    genes = tissue[0].columns[3:]\n",
    "\n",
    "    for gene in genes:\n",
    "      y = tissue[0][gene].to_list()\n",
    "      r_value, p_value = scipy.stats.spearmanr(x, y)\n",
    "\n",
    "      if r_value > 0.65:\n",
    "        genepos.append(gene)\n",
    "        rvaluepos.append(r_value)\n",
    "        pvaluepos.append(p_value)\n",
    "        \n",
    "      elif r_value < -0.65:\n",
    "        geneneg.append(gene)\n",
    "        rvalueneg.append(r_value)\n",
    "        pvalueneg.append(p_value)\n",
    "\n",
    "    #create a dictionnary\n",
    "    d = dict(ensembl_gene_id= genepos, r_value=rvaluepos,  p_value=pvaluepos)\n",
    "    d2 = dict(Gene= geneneg, r_value=rvalueneg,  p_value=pvalueneg)\n",
    "\n",
    "    #Convert to dataframe\n",
    "    positiv_corr = 'pos_corr_'+tissue[1]\n",
    "    negativ_corr = 'neg_corr_'+tissue[1]\n",
    "\n",
    "    df_positiv_corr = pandas.DataFrame.from_dict(d, orient='index')\n",
    "    df_negativ_corr = pandas.DataFrame.from_dict(data=d2, orient='index')\n",
    "\n",
    "    #Transpose to have a better table\n",
    "    df_positiv_corr = df_positiv_corr.transpose()\n",
    "    df_negativ_corr = df_negativ_corr.transpose()\n",
    "\n",
    "    #name\n",
    "    positiv_corr = 'pos_spearman_corr_'+tissue[1]+'_'+isoform\n",
    "    negativ_corr = 'neg_spearman_corr_'+tissue[1]+'_'+isoform\n",
    "\n",
    "    print(positiv_corr, df_positiv_corr.shape)\n",
    "    print(negativ_corr, df_negativ_corr.shape)\n",
    "\n",
    "    #scending orde, save to excel\n",
    "    df_positiv_corr['info'] = tissue[1]+'_'+isoform\n",
    "    df_positiv_corr = protein_coding_list.merge(df_positiv_corr)\n",
    "    df_positiv_corr = df_positiv_corr.drop(['entrezgene_id', 'transcript_biotype'], axis = 1)\n",
    "\n",
    "    df_positiv_corr = df_positiv_corr.sort_values(['r_value'], ascending = False)\n",
    "\n",
    "    df_positiv_corr.to_excel(positiv_corr+'.xlsx', index=False)\n",
    "    df_negativ_corr.to_excel(negativ_corr+'.xlsx', index=False)\n",
    "\n",
    "    all_positiv_corr_spearman.append(df_positiv_corr)\n",
    "  print()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "id": "OGqyJk-14Zyq"
   },
   "outputs": [],
   "source": [
    "#Graphic representation, two graphs in one\n",
    "\n",
    "all_tissue_used = [[df_normal_exp,'Normal Tissues'], [df_primary_exp,'Primary Tumors']]\n",
    "\n",
    "HK2 = 'ENSG00000159399'\n",
    "YAP1 = 'ENSG00000137693'\n",
    "AJUBA = 'ENSG00000129474'\n",
    "YBX1 = 'ENSG00000065978'\n",
    "TGFB1 = 'ENSG00000105329'\n",
    "MAPK1 = 'ENSG00000100030'\n",
    "MTDH = 'ENSG00000147649'\n",
    "KPNA1 = 'ENSG00000114030'\n",
    "TNPO1 = 'ENSG00000083312'\n",
    "\n",
    "\n",
    " \n",
    "genes = [[ATG7,'ATG7'], [ATG7_1,'ATG7_1'], [MAPK1,'MAPK1'], [MTDH,'MTDH'], [KPNA1,'KPNA1'], \n",
    "         [TNPO1,'TNPO1'], [HK2,'HK2'], [YAP1,'YAP1'], [AJUBA,'AJUBA'], [YBX1,'YBX1'], [TGFB1,'TGFB1']]\n",
    "\n",
    "for gene in genes:\n",
    "  for df in data:\n",
    "    \n",
    "      fig, ax = plt.subplots(1,2, figsize=(18,8))\n",
    "      name = gene[1]+ \" correlation with the two isoforms\"\n",
    "      name2 = gene[1]+ \" expression (log2 tpm)\"\n",
    "\n",
    "      fig.suptitle(name, fontsize=\"x-large\")\n",
    "\n",
    "      x = df[0][gene[0]].to_list()\n",
    "      y1 = df[0]['ATG7_1'].to_list()\n",
    "      y2 = df[0]['ATG7_2'].to_list()\n",
    "\n",
    "\n",
    "      fig = seaborn.regplot(x=x,y=y1, ax=ax[0], x_bins=500, label = df[1], color = df[2])\n",
    "      fig.set_title(\"ATG7(1)\")\n",
    "      fig.set_xlabel(name2)\n",
    "      fig.set_ylabel('ATG7(1) expression (log2 tpm)')\n",
    "\n",
    "      fig = seaborn.regplot(x=x,y=y2, ax=ax[1], x_bins=500, label = df[1], color = df[2])\n",
    "      fig.set_title(\"ATG7(2)\")\n",
    "      fig.set_xlabel(name2)\n",
    "      fig.set_ylabel('ATG7(2) expression (log2 tpm)')\n",
    "      plt.legend()\n",
    "      plt.show()\n",
    "      plt.clf()\n",
    "\n",
    "      print(df[1])\n",
    "      r_value, p_value = scipy.stats.spearmanr(x, y1)\n",
    "      print('  correlation between ATG7_1 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "      r_value, p_value = scipy.stats.spearmanr(x, y2)\n",
    "      print('  correlation between ATG7_2 and', gene[1],  'is:', 'r_value = {:.2f} for a p_value of {:.2e}'.format(r_value, p_value))\n",
    "      print()"
   ]
  }
 ],
 "metadata": {
  "colab": {
   "collapsed_sections": [],
   "name": "all_organs.ipynb",
   "provenance": [],
   "toc_visible": true
  },
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.11"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 1
}
